diff --git a/templates/cart/_cartList.html.twig b/templates/cart/_cartList.html.twig
new file mode 100644
index 0000000..5da81fc
--- /dev/null
+++ b/templates/cart/_cartList.html.twig
@@ -0,0 +1,75 @@
+<div>
+    <div class="row p-3">
+        <div class="col-3">
+            Item Name
+        </div>
+        <div class="col-3">
+            Quantity
+        </div>
+        <div class="col-3">
+            Price
+        </div>
+        <div class="col-3"></div>
+    </div>
+    {% for item in cart.items %}
+        <div
+            class="cart-item row p-3"
+            data-action="submit-confirm:async:submitted->cart-list#removeItem"
+        >
+            <div class="col-2">
+                {{ item.product.name }}
+            </div>
+            <div class="col-1">
+                {% if item.color %}
+                    <span class="color-square" style="background-color: rgb({{ item.color.red }}, {{ item.color.green }}, {{ item.color.blue }});"></span>
+                {% endif %}
+            </div>
+            <div class="col-3">
+                {{ item.quantity }}
+            </div>
+            <div class="col-3">
+                {{ item.totalString|format_currency('USD') }}
+            </div>
+            <div class="col-3">
+                <form
+                    action="{{ path('app_cart_remove_item', {
+                        productId: item.product.id,
+                        colorId: item.color ? item.color.id: null,
+                    }) }}"
+                    method="POST"
+                    {{ stimulus_controller('submit-confirm', {
+                        title: 'Remove this item?',
+                        icon: 'warning',
+                        confirmButtonText: 'Yes, remove it',
+                        submitAsync: true,
+                    }) }}
+                    data-action="submit-confirm#onSubmit"
+                >
+                    <input type="hidden" name="_token" value="{{ csrf_token('remove_item') }}" />
+
+                    <button class="btn btn-info btn-sm">
+                        Remove
+                    </button>
+                </form>
+            </div>
+        </div>
+    {% else %}
+        <div class="col-12">
+            Your cart is empty! Get to shopping!
+        </div>
+    {% endfor %}
+
+    {% if cart.items|length > 0 %}
+    <div class="p-3">
+        Total: <strong>{{ cart.totalString|format_currency('USD') }}</strong>
+    </div>
+    {% endif %}
+</div>
+
+{% if cart.items|length > 0 %}
+<div>
+    <a href="{{ path('app_checkout') }}" class="btn btn-primary">
+        Check Out &gt;&gt;
+    </a>
+</div>
+{% endif %}
diff --git a/templates/cart/cart.html.twig b/templates/cart/cart.html.twig
index 8868958..4822828 100644
--- a/templates/cart/cart.html.twig
+++ b/templates/cart/cart.html.twig
@@ -15,81 +15,7 @@
                     class="component-light p-3"
                     {{ stimulus_controller('cart-list') }}
                 >
-                    <div>
-                        <div class="row p-3">
-                            <div class="col-3">
-                                Item Name
-                            </div>
-                            <div class="col-3">
-                                Quantity
-                            </div>
-                            <div class="col-3">
-                                Price
-                            </div>
-                            <div class="col-3"></div>
-                        </div>
-                        {% for item in cart.items %}
-                            <div
-                                class="cart-item row p-3"
-                                data-action="submit-confirm:async:submitted->cart-list#removeItem"
-                            >
-                                <div class="col-2">
-                                    {{ item.product.name }}
-                                </div>
-                                <div class="col-1">
-                                    {% if item.color %}
-                                        <span class="color-square" style="background-color: rgb({{ item.color.red }}, {{ item.color.green }}, {{ item.color.blue }});"></span>
-                                    {% endif %}
-                                </div>
-                                <div class="col-3">
-                                    {{ item.quantity }}
-                                </div>
-                                <div class="col-3">
-                                    {{ item.totalString|format_currency('USD') }}
-                                </div>
-                                <div class="col-3">
-                                    <form
-                                        action="{{ path('app_cart_remove_item', {
-                                            productId: item.product.id,
-                                            colorId: item.color ? item.color.id: null,
-                                        }) }}"
-                                        method="POST"
-                                        {{ stimulus_controller('submit-confirm', {
-                                            title: 'Remove this item?',
-                                            icon: 'warning',
-                                            confirmButtonText: 'Yes, remove it',
-                                            submitAsync: true,
-                                        }) }}
-                                        data-action="submit-confirm#onSubmit"
-                                    >
-                                        <input type="hidden" name="_token" value="{{ csrf_token('remove_item') }}" />
-
-                                        <button class="btn btn-info btn-sm">
-                                            Remove
-                                        </button>
-                                    </form>
-                                </div>
-                            </div>
-                        {% else %}
-                            <div class="col-12">
-                                Your cart is empty! Get to shopping!
-                            </div>
-                        {% endfor %}
-
-                        {% if cart.items|length > 0 %}
-                        <div class="p-3">
-                            Total: <strong>{{ cart.totalString|format_currency('USD') }}</strong>
-                        </div>
-                        {% endif %}
-                    </div>
-
-                    {% if cart.items|length > 0 %}
-                    <div>
-                        <a href="{{ path('app_checkout') }}" class="btn btn-primary">
-                            Check Out &gt;&gt;
-                        </a>
-                    </div>
-                    {% endif %}
+                    {{ include('cart/_cartList.html.twig') }}
                 </div>
             </div>
         </div>
