diff --git a/assets/controllers/submit-confirm_controller.js b/assets/controllers/submit-confirm_controller.js
index c28004cfa..ef3849e56 100644
--- a/assets/controllers/submit-confirm_controller.js
+++ b/assets/controllers/submit-confirm_controller.js
@@ -12,7 +12,7 @@ export default class extends Controller {
     }
 
     connect() {
-        useDispatch(this, { debug: true });
+        useDispatch(this);
     }
 
     onSubmit(event) {
