diff --git a/assets/controllers/submit-confirm_controller.js b/assets/controllers/submit-confirm_controller.js
index be60b0ae..c28004cf 100644
--- a/assets/controllers/submit-confirm_controller.js
+++ b/assets/controllers/submit-confirm_controller.js
@@ -1,5 +1,6 @@
 import { Controller } from 'stimulus';
 import Swal from 'sweetalert2';
+import { useDispatch } from 'stimulus-use';
 
 export default class extends Controller {
     static values = {
@@ -10,6 +11,10 @@ export default class extends Controller {
         submitAsync: Boolean,
     }
 
+    connect() {
+        useDispatch(this, { debug: true });
+    }
+
     onSubmit(event) {
         event.preventDefault();
 
@@ -28,16 +33,20 @@ export default class extends Controller {
         });
     }
 
-    submitForm() {
+    async submitForm() {
         if (!this.submitAsyncValue) {
             this.element.submit();
 
             return;
         }
 
-        return fetch(this.element.action, {
+        const response = await fetch(this.element.action, {
             method: this.element.method,
             body: new URLSearchParams(new FormData(this.element)),
         });
+
+        this.dispatch('async:submitted', {
+            response,
+        })
     }
 }
