diff --git a/templates/product/_searchPreview.html.twig b/templates/product/_searchPreview.html.twig
index 7d1b0b9b9..37faae279 100644
--- a/templates/product/_searchPreview.html.twig
+++ b/templates/product/_searchPreview.html.twig
@@ -1,6 +1,7 @@
 <div class="list-group">
     {% for product in products %}
         <a
+            role="option"
             class="list-group-item list-group-item-action"
             href="{{ path('app_product', { id: product.id }) }}"
         >
@@ -17,6 +18,6 @@
             </div>
         </a>
     {% else %}
-        <div class="list-group-item">No results found!</div>
+        <div class="list-group-item" role="option" aria-disabled="true">No results found!</div>
     {% endfor %}
 </div>
