diff --git a/templates/product/index.html.twig b/templates/product/index.html.twig
index cf3265a..13840ad 100644
--- a/templates/product/index.html.twig
+++ b/templates/product/index.html.twig
@@ -39,7 +39,7 @@
                             <div
                                 class="input-group"
                                 {{ stimulus_controller('autocomplete', {
-                                    url: path('app_homepage')
+                                    url: path('app_homepage', { preview: 1 })
                                 }) }}
                             >
                                 <input
