diff --git a/assets/controllers/counter_controller.js b/assets/controllers/counter_controller.js
index b8148597a..e9fd157ad 100644
--- a/assets/controllers/counter_controller.js
+++ b/assets/controllers/counter_controller.js
@@ -1,12 +1,9 @@
 import { Controller } from 'stimulus';
 
 export default class extends Controller {
+    count = 0;
     static targets = ['count'];
 
-    connect() {
-        this.count = 0;
-    }
-
     increment() {
         this.count++;
         this.countTarget.innerText = this.count;
