diff --git a/templates/product/index.html.twig b/templates/product/index.html.twig
index eb7d337..90b523f 100644
--- a/templates/product/index.html.twig
+++ b/templates/product/index.html.twig
@@ -10,6 +10,12 @@
             <div class="col-xs-12 col-9">
 
                 <div data-controller="counter">
+                    <button class="btn btn-primary btn-sm">
+                        Click me!
+                    </button>
+
+                    <br>
+
                     I have been clicked
                     <span data-counter-target="count">0</span>
                     times!
