diff --git a/assets/controllers/autocomplete-transition_controller.js b/assets/controllers/autocomplete-transition_controller.js
index 10b530e0a..2ff555036 100644
--- a/assets/controllers/autocomplete-transition_controller.js
+++ b/assets/controllers/autocomplete-transition_controller.js
@@ -9,6 +9,10 @@ export default class extends Controller {
     }
 
     toggle(event) {
-        console.log(event);
+        if (event.detail.action === 'open') {
+            this.enter();
+        } else {
+            this.leave();
+        }
     }
 }
