diff --git a/assets/controllers/autocomplete-transition_controller.js b/assets/controllers/autocomplete-transition_controller.js
index e54a46e6..63e48c67 100644
--- a/assets/controllers/autocomplete-transition_controller.js
+++ b/assets/controllers/autocomplete-transition_controller.js
@@ -4,4 +4,8 @@ export default class extends Controller {
     connect() {
         console.log('I want transitions!');
     }
+
+    toggle(event) {
+        console.log(event);
+    }
 }
diff --git a/templates/product/index.html.twig b/templates/product/index.html.twig
index e88a98e8..8ca16d74 100644
--- a/templates/product/index.html.twig
+++ b/templates/product/index.html.twig
@@ -44,6 +44,7 @@
                                     },
                                     'autocomplete-transition': {}
                                 }) }}
+                                data-action="toggle->autocomplete-transition#toggle"
                             >
                                 <input
                                     name="q"
