diff --git a/assets/controllers/modal-form_controller.js b/assets/controllers/modal-form_controller.js
index 8b512139d..85873baa3 100644
--- a/assets/controllers/modal-form_controller.js
+++ b/assets/controllers/modal-form_controller.js
@@ -19,10 +19,16 @@ export default class extends Controller {
     async submitForm(event) {
         event.preventDefault();
         const $form = $(this.modalBodyTarget).find('form');
-        this.modalBodyTarget.innerHTML = await $.ajax({
-            url: this.formUrlValue,
-            method: $form.prop('method'),
-            data: $form.serialize(),
-        });
+
+        try {
+            await $.ajax({
+                url: this.formUrlValue,
+                method: $form.prop('method'),
+                data: $form.serialize(),
+            });
+            console.log('success!');
+        } catch (e) {
+            this.modalBodyTarget.innerHTML = e.responseText;
+        }
     }
 }
