diff --git a/assets/controllers/modal-form_controller.js b/assets/controllers/modal-form_controller.js
index 85873ba..a59fd79 100644
--- a/assets/controllers/modal-form_controller.js
+++ b/assets/controllers/modal-form_controller.js
@@ -7,11 +7,12 @@ export default class extends Controller {
     static values = {
         formUrl: String,
     }
+    modal = null;
 
     async openModal(event) {
         this.modalBodyTarget.innerHTML = 'Loading...';
-        const modal = new Modal(this.modalTarget);
-        modal.show();
+        this.modal = new Modal(this.modalTarget);
+        this.modal.show();
 
         this.modalBodyTarget.innerHTML = await $.ajax(this.formUrlValue);
     }
@@ -26,7 +27,7 @@ export default class extends Controller {
                 method: $form.prop('method'),
                 data: $form.serialize(),
             });
-            console.log('success!');
+            this.modal.hide();
         } catch (e) {
             this.modalBodyTarget.innerHTML = e.responseText;
         }
