diff --git a/assets/controllers/modal-form_controller.js b/assets/controllers/modal-form_controller.js
index a59fd79f..5267dd52 100644
--- a/assets/controllers/modal-form_controller.js
+++ b/assets/controllers/modal-form_controller.js
@@ -32,4 +32,8 @@ export default class extends Controller {
             this.modalBodyTarget.innerHTML = e.responseText;
         }
     }
+
+    modalHidden() {
+        console.log('it was hidden!');
+    }
 }
diff --git a/templates/product_admin/index.html.twig b/templates/product_admin/index.html.twig
index 93d133ff..0727b11a 100644
--- a/templates/product_admin/index.html.twig
+++ b/templates/product_admin/index.html.twig
@@ -7,9 +7,12 @@
     <div class="d-flex flex-row">
         <h1 class="me-3">Product index</h1>
 
-        <div {{ stimulus_controller('modal-form', {
-            formUrl: path('product_admin_new')
-        }) }}>
+        <div
+            {{ stimulus_controller('modal-form', {
+                formUrl: path('product_admin_new')
+            }) }}
+            data-action="hidden.bs.modal->modal-form#modalHidden"
+        >
             <button
                 class="btn btn-primary btn-sm"
                 data-action="modal-form#openModal"
