diff --git a/assets/controllers/admin-chartjs_controller.js b/assets/controllers/admin-chartjs_controller.js
index bdb18309..20ee8b19 100644
--- a/assets/controllers/admin-chartjs_controller.js
+++ b/assets/controllers/admin-chartjs_controller.js
@@ -2,6 +2,15 @@ import { Controller } from 'stimulus';
 
 export default class extends Controller {
     onChartConnect(event) {
-        console.log(event);
+        this.chart = event.detail.chart;
+
+        setTimeout(() => {
+            this.setNewData();
+        }, 5000)
+    }
+
+    setNewData() {
+        this.chart.data.datasets[0].data[2] = 30;
+        this.chart.update();
     }
 }
