diff --git a/src/Controller/AdminController.php b/src/Controller/AdminController.php
index 2cf2b53d..c3f871dd 100644
--- a/src/Controller/AdminController.php
+++ b/src/Controller/AdminController.php
@@ -4,14 +4,31 @@ namespace App\Controller;
 
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\Routing\Annotation\Route;
+use Symfony\UX\Chartjs\Builder\ChartBuilderInterface;
+use Symfony\UX\Chartjs\Model\Chart;
 
 class AdminController extends AbstractController
 {
     /**
      * @Route("/admin", name="admin_dashboard")
      */
-    public function dashboard()
+    public function dashboard(ChartBuilderInterface $chartBuilder)
     {
-        return $this->render('admin/dashboard.html.twig');
+        $chart = $chartBuilder->createChart(Chart::TYPE_LINE);
+        $chart->setData([
+            'labels' => ['January', 'February', 'March', 'April', 'May', 'June', 'July'],
+            'datasets' => [
+                [
+                    'label' => 'Sales!',
+                    'backgroundColor' => 'rgb(255, 99, 132)',
+                    'borderColor' => 'rgb(255, 99, 132)',
+                    'data' => [522, 1500, 2250, 2197, 2345, 3122, 3099],
+                ],
+            ],
+        ]);
+
+        return $this->render('admin/dashboard.html.twig', [
+            'chart' => $chart
+        ]);
     }
 }
diff --git a/templates/admin/dashboard.html.twig b/templates/admin/dashboard.html.twig
index fa26e2c4..f3a426a1 100644
--- a/templates/admin/dashboard.html.twig
+++ b/templates/admin/dashboard.html.twig
@@ -101,6 +101,8 @@
 
             <div class="col-10 mt-4">
                 <h1>Admin Dashboard</h1>
+
+                {{ render_chart(chart) }}
             </div>
         </div>
     </div>
