diff --git a/assets/controllers/admin-chartjs_controller.js b/assets/controllers/admin-chartjs_controller.js
index 8a35b87..505064f 100644
--- a/assets/controllers/admin-chartjs_controller.js
+++ b/assets/controllers/admin-chartjs_controller.js
@@ -2,6 +2,12 @@ import { Controller } from 'stimulus';
 
 export default class extends Controller {
     connect() {
-        console.log('📈');
+        this.element.addEventListener('chartjs:connect', (event) => {
+            this.onChartConnect(event);
+        });
+    }
+
+    onChartConnect(event) {
+        console.log(event);
     }
 }
