diff --git a/src/Controller/AdminController.php b/src/Controller/AdminController.php
index c3f871dd4..db1ec01f0 100644
--- a/src/Controller/AdminController.php
+++ b/src/Controller/AdminController.php
@@ -26,6 +26,15 @@ class AdminController extends AbstractController
                 ],
             ],
         ]);
+        $chart->setOptions([
+            'scales' => [
+                'yAxes' => [[
+                    'ticks' => [
+                        'beginAtZero' => true
+                    ]
+                ]]
+            ]
+        ]);
 
         return $this->render('admin/dashboard.html.twig', [
             'chart' => $chart
