diff --git a/assets/controllers/admin-chartjs_controller.js b/assets/controllers/admin-chartjs_controller.js
new file mode 100644
index 000000000..8a35b8705
--- /dev/null
+++ b/assets/controllers/admin-chartjs_controller.js
@@ -0,0 +1,7 @@
+import { Controller } from 'stimulus';
+
+export default class extends Controller {
+    connect() {
+        console.log('📈');
+    }
+}
diff --git a/templates/admin/dashboard.html.twig b/templates/admin/dashboard.html.twig
index f3a426a16..a8cacec4e 100644
--- a/templates/admin/dashboard.html.twig
+++ b/templates/admin/dashboard.html.twig
@@ -102,7 +102,7 @@
             <div class="col-10 mt-4">
                 <h1>Admin Dashboard</h1>
 
-                {{ render_chart(chart) }}
+                {{ render_chart(chart, {'data-controller': 'admin-chartjs'}) }}
             </div>
         </div>
     </div>
