diff --git a/assets/controllers/color-square_controller.js b/assets/controllers/color-square_controller.js
index 0b8aacd..49051f0 100644
--- a/assets/controllers/color-square_controller.js
+++ b/assets/controllers/color-square_controller.js
@@ -2,6 +2,6 @@ import { Controller } from 'stimulus';
 
 export default class extends Controller {
     selectColor(event) {
-        console.log(event);
+        event.currentTarget.classList.add('selected');
     }
 }
