diff --git a/templates/product/_cart_add_controls.html.twig b/templates/product/_cart_add_controls.html.twig
index 1f687ae0..8966036d 100644
--- a/templates/product/_cart_add_controls.html.twig
+++ b/templates/product/_cart_add_controls.html.twig
@@ -3,6 +3,10 @@
 }) }}
     {% if addToCartForm.color is defined %}
         {{ form_widget(addToCartForm.color) }}
+
+        {% for color in addToCartForm.vars.data.product.colors %}
+            <div>#{{ color.hexColor }}</div>
+        {% endfor %}
     {% endif %}
 
     {{ form_widget(addToCartForm.quantity, {
