diff --git a/templates/product/_cart_add_controls.html.twig b/templates/product/_cart_add_controls.html.twig
index bc7d58cbb..60b40290e 100644
--- a/templates/product/_cart_add_controls.html.twig
+++ b/templates/product/_cart_add_controls.html.twig
@@ -8,7 +8,7 @@
             {% for color in addToCartForm.vars.data.product.colors %}
                 <span
                     class="color-square"
-                    data-action="color-square#selectColor"
+                    data-action="click->color-square#selectColor"
                     style="background-color: rgb({{ color.red }}, {{ color.green }}, {{ color.blue }});"
                 ></span>
             {% endfor %}
