diff --git a/assets/controllers/color-square_controller.js b/assets/controllers/color-square_controller.js
index 6a1cac9b..2449e7cd 100644
--- a/assets/controllers/color-square_controller.js
+++ b/assets/controllers/color-square_controller.js
@@ -1,7 +1,7 @@
 import { Controller } from 'stimulus';
 
 export default class extends Controller {
-    targets = ['colorSquare']
+    static targets = ['colorSquare']
 
     selectColor(event) {
         console.log(this.colorSquareTargets);
