diff --git a/templates/base.html.twig b/templates/base.html.twig
index 410f7190..c0d40094 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -25,7 +25,7 @@
                         <span class="p-3">MVP Office Supplies</span>
                     </a>
 
-                    <ul class="navbar-nav">
+                    <ul class="navbar-nav" data-turbo="true">
                         <li class="nav-item">
                             <a class="nav-link" href="{{ path('app_cart') }}">
                                 Shopping Cart ({{ count_cart_items() }})
