diff --git a/templates/base.html.twig b/templates/base.html.twig
index 5d6dcb2b..410f7190 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -11,7 +11,7 @@
             {{ encore_entry_script_tags('app') }}
         {% endblock %}
     </head>
-    <body>
+    <body data-turbo="false">
         <div class="page-top">
             <header class="header px-2">
                 <nav class="navbar navbar-expand-lg navbar-dark justify-content-between">
