diff --git a/templates/broadcast/Review.stream.html.twig b/templates/broadcast/Review.stream.html.twig
index a3f26596e..d1d1c973c 100644
--- a/templates/broadcast/Review.stream.html.twig
+++ b/templates/broadcast/Review.stream.html.twig
@@ -28,5 +28,5 @@
 {% endblock %}
 
 {% block remove %}
-    REMOVE!
+    <turbo-stream action="remove" target="product-review-{{ id }}"></turbo-stream>
 {% endblock %}
