diff --git a/src/Entity/Review.php b/src/Entity/Review.php
index f41106219..659cd2923 100644
--- a/src/Entity/Review.php
+++ b/src/Entity/Review.php
@@ -5,9 +5,11 @@ namespace App\Entity;
 use App\Repository\ReviewRepository;
 use Doctrine\ORM\Mapping as ORM;
 use Symfony\Component\Validator\Constraints as Assert;
+use Symfony\UX\Turbo\Attribute\Broadcast;
 
 /**
  * @ORM\Entity(repositoryClass=ReviewRepository::class)
+ * @Broadcast()
  */
 class Review
 {
