diff --git a/src/EventSubscriber/TurboFrameRedirectSubscriber.php b/src/EventSubscriber/TurboFrameRedirectSubscriber.php
index 9bfdb04..c17c437 100644
--- a/src/EventSubscriber/TurboFrameRedirectSubscriber.php
+++ b/src/EventSubscriber/TurboFrameRedirectSubscriber.php
@@ -6,9 +6,17 @@ use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\HttpKernel\Event\ResponseEvent;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
 
 class TurboFrameRedirectSubscriber implements EventSubscriberInterface
 {
+    private UrlGeneratorInterface $urlGenerator;
+
+    public function __construct(UrlGeneratorInterface $urlGenerator)
+    {
+        $this->urlGenerator = $urlGenerator;
+    }
+
     public function onKernelResponse(ResponseEvent $event)
     {
         if (!$this->shouldWrapRedirect($event->getRequest(), $event->getResponse())) {
@@ -42,6 +50,12 @@ class TurboFrameRedirectSubscriber implements EventSubscriberInterface
             return true;
         }
 
+        $location = $response->headers->get('Location');
+
+        if ($location === $this->urlGenerator->generate('app_login')) {
+            return true;
+        }
+
         return false;
     }
 }
