diff --git a/assets/turbo/turbo-helper.js b/assets/turbo/turbo-helper.js
index 4324eb2..54dedb0 100644
--- a/assets/turbo/turbo-helper.js
+++ b/assets/turbo/turbo-helper.js
@@ -122,22 +122,14 @@ const TurboHelper = class {
 
     beforeFetchResponse(event) {
         const fetchResponse = event.detail.fetchResponse;
-        if (!fetchResponse.succeeded || !fetchResponse.redirected) {
-            return;
-        }
-
-        if (!this.getCurrentFrame() || !this.getCurrentFrame().dataset.turboFormRedirect) {
+        const redirectLocation = fetchResponse.response.headers.get('Turbo-Location');
+        if (!redirectLocation) {
             return;
         }
 
         event.preventDefault();
         Turbo.clearCache();
-
-        Turbo.visit(fetchResponse.location);
-    }
-
-    getCurrentFrame() {
-        return document.querySelector('turbo-frame[busy]');
+        Turbo.visit(redirectLocation);
     }
 }
 
