diff --git a/src/EventSubscriber/TurboFrameRedirectSubscriber.php b/src/EventSubscriber/TurboFrameRedirectSubscriber.php
index eee47c67..9bfdb042 100644
--- a/src/EventSubscriber/TurboFrameRedirectSubscriber.php
+++ b/src/EventSubscriber/TurboFrameRedirectSubscriber.php
@@ -14,6 +14,11 @@ class TurboFrameRedirectSubscriber implements EventSubscriberInterface
         if (!$this->shouldWrapRedirect($event->getRequest(), $event->getResponse())) {
             return;
         }
+
+        $response = new Response(null, 200, [
+            'Turbo-Location' => $event->getResponse()->headers->get('Location'),
+        ]);
+        $event->setResponse($response);
     }
 
     public static function getSubscribedEvents()
