diff --git a/src/Controller/ProductAdminController.php b/src/Controller/ProductAdminController.php
index 4d16bb1c5..2f347d20b 100644
--- a/src/Controller/ProductAdminController.php
+++ b/src/Controller/ProductAdminController.php
@@ -75,10 +75,10 @@ class ProductAdminController extends AbstractController
             return $this->redirectToRoute('product_admin_index');
         }
 
-        return $this->render('product_admin/edit.html.twig', [
+        return $this->renderForm('product_admin/edit.html.twig', [
             'product' => $product,
-            'form' => $form->createView(),
-        ], new Response(null, $form->isSubmitted() && !$form->isValid() ? 422 : 200));
+            'form' => $form,
+        ]);
     }
 
     /**
