diff --git a/src/Controller/ProductAdminController.php b/src/Controller/ProductAdminController.php
index fbc0ba27f..4d16bb1c5 100644
--- a/src/Controller/ProductAdminController.php
+++ b/src/Controller/ProductAdminController.php
@@ -78,7 +78,7 @@ class ProductAdminController extends AbstractController
         return $this->render('product_admin/edit.html.twig', [
             'product' => $product,
             'form' => $form->createView(),
-        ]);
+        ], new Response(null, $form->isSubmitted() && !$form->isValid() ? 422 : 200));
     }
 
     /**
