diff --git a/assets/turbo/turbo-helper.js b/assets/turbo/turbo-helper.js
index fe25b14..dc6817f 100644
--- a/assets/turbo/turbo-helper.js
+++ b/assets/turbo/turbo-helper.js
@@ -102,15 +102,17 @@ const TurboHelper = class {
     }
 
     beforeFetchResponse(event) {
-        if (!this.modal || !this.modal._isShown) {
+        const fetchResponse = event.detail.fetchResponse;
+        if (!fetchResponse.succeeded || !fetchResponse.redirected) {
             return;
         }
 
-        const fetchResponse = event.detail.fetchResponse;
-        if (fetchResponse.succeeded && fetchResponse.redirected) {
-            event.preventDefault();
-            Turbo.visit(fetchResponse.location);
+        if (!this.getCurrentFrame() || !this.getCurrentFrame().dataset.turboFormRedirect) {
+            return;
         }
+
+        event.preventDefault();
+        Turbo.visit(fetchResponse.location);
     }
 
     getCurrentFrame() {
