diff --git a/templates/product/_cart_add_controls.html.twig b/templates/product/_cart_add_controls.html.twig
index f0deedd..774f929 100644
--- a/templates/product/_cart_add_controls.html.twig
+++ b/templates/product/_cart_add_controls.html.twig
@@ -1,3 +1,4 @@
+<turbo-frame id="add-to-cart-controls">
 {{ form_start(addToCartForm, {
     attr: { class: 'cart-add-controls d-flex align-items-center justify-content-baseline' }
 }) }}
@@ -40,3 +41,4 @@
     {% endif %}
     {{ form_errors(addToCartForm.quantity) }}
 </div>
+</turbo-frame>
