diff --git a/assets/styles/app.css b/assets/styles/app.css
index 624e9f141..c92cf621f 100644
--- a/assets/styles/app.css
+++ b/assets/styles/app.css
@@ -20,7 +20,7 @@ turbo-frame {
     display: block;
 }
 turbo-frame[busy] {
-    opacity: .2;
+    opacity: .7;
 }
 turbo-frame[busy] .frame-loading-hide, turbo-frame .frame-loading-show {
     display: none;
diff --git a/src/Controller/CartController.php b/src/Controller/CartController.php
index f7d055abb..b32cfe2d3 100644
--- a/src/Controller/CartController.php
+++ b/src/Controller/CartController.php
@@ -38,8 +38,6 @@ class CartController extends AbstractController
             'product' => $featuredProduct,
         ]);
 
-        sleep(3);
-
         return $this->renderForm('cart/_featuredSidebar.html.twig', [
             'featuredProduct' => $featuredProduct,
             'addToCartForm' => $addToCartForm,
