diff --git a/assets/styles/app.css b/assets/styles/app.css
index 4808859e7..bc83f84e0 100644
--- a/assets/styles/app.css
+++ b/assets/styles/app.css
@@ -16,7 +16,12 @@ body.turbo-loading {
     opacity: .2;
 }
 */
-
+turbo-frame[busy] .frame-loading-hide, turbo-frame .frame-loading-show {
+    display: none;
+}
+turbo-frame[busy] .frame-loading-show {
+    display: inline-block;
+}
 a, a:hover  {
     color: #000;
 }
diff --git a/templates/cart/_featuredSidebar.html.twig b/templates/cart/_featuredSidebar.html.twig
index 5bf1009d4..113df23c5 100644
--- a/templates/cart/_featuredSidebar.html.twig
+++ b/templates/cart/_featuredSidebar.html.twig
@@ -22,9 +22,11 @@
 
                 <a
                     data-turbo-frame="cart-sidebar"
+                    class="frame-loading-hide"
                     href="{{ path('_app_cart_product_featured', {
                     description: true,
                 }) }}">(read more)</a>
+                <span class="frame-loading-show fas fa-spinner fa-spin"></span>
             {% endif %}
 
             {{ include('product/_cart_add_controls.html.twig') }}
