diff --git a/templates/_modal.html.twig b/templates/_modal.html.twig
index f907c361..62bf84d3 100644
--- a/templates/_modal.html.twig
+++ b/templates/_modal.html.twig
@@ -17,6 +17,7 @@
                 src="{{ modalSrc }}"
                 id="{{ id }}"
                 loading="lazy"
+                data-turbo-form-redirect="true"
             >
                 {{ modalContent|default('Loading...') }}
             </turbo-frame>
