diff --git a/assets/styles/app.css b/assets/styles/app.css
index 4808859..561c643 100644
--- a/assets/styles/app.css
+++ b/assets/styles/app.css
@@ -16,6 +16,9 @@ body.turbo-loading {
     opacity: .2;
 }
 */
+turbo-frame[busy] {
+    opacity: .2;
+}
 
 a, a:hover  {
     color: #000;
diff --git a/src/Controller/CartController.php b/src/Controller/CartController.php
index 778dd96..6a3ec75 100644
--- a/src/Controller/CartController.php
+++ b/src/Controller/CartController.php
@@ -40,6 +40,8 @@ class CartController extends AbstractController
             'product' => $featuredProduct,
         ]);
 
+        sleep(5);
+
         return $this->renderForm('cart/cart.html.twig', [
             'featuredProduct' => $featuredProduct,
             'addToCartForm' => $addToCartForm,
