diff --git a/assets/styles/app.css b/assets/styles/app.css
index ca74f75..2167fbb 100644
--- a/assets/styles/app.css
+++ b/assets/styles/app.css
@@ -20,7 +20,7 @@ turbo-frame {
     display: block;
 }
 turbo-frame[busy] {
-    opacity: .2;
+    opacity: .7;
 }
 
 a, a:hover  {
diff --git a/src/Controller/CartController.php b/src/Controller/CartController.php
index 6a3ec75..778dd96 100644
--- a/src/Controller/CartController.php
+++ b/src/Controller/CartController.php
@@ -40,8 +40,6 @@ class CartController extends AbstractController
             'product' => $featuredProduct,
         ]);
 
-        sleep(5);
-
         return $this->renderForm('cart/cart.html.twig', [
             'featuredProduct' => $featuredProduct,
             'addToCartForm' => $addToCartForm,
