diff --git a/src/Controller/CartController.php b/src/Controller/CartController.php
index b32cfe2..f7d055a 100644
--- a/src/Controller/CartController.php
+++ b/src/Controller/CartController.php
@@ -38,6 +38,8 @@ class CartController extends AbstractController
             'product' => $featuredProduct,
         ]);
 
+        sleep(3);
+
         return $this->renderForm('cart/_featuredSidebar.html.twig', [
             'featuredProduct' => $featuredProduct,
             'addToCartForm' => $addToCartForm,
