diff --git a/assets/turbo/turbo-helper.js b/assets/turbo/turbo-helper.js
index dc6817f2..c5e9bcfb 100644
--- a/assets/turbo/turbo-helper.js
+++ b/assets/turbo/turbo-helper.js
@@ -112,6 +112,7 @@ const TurboHelper = class {
         }
 
         event.preventDefault();
+        Turbo.clearCache();
         Turbo.visit(fetchResponse.location);
     }
 
diff --git a/templates/product/_cart_add_controls.html.twig b/templates/product/_cart_add_controls.html.twig
index 774f929d..dfd01668 100644
--- a/templates/product/_cart_add_controls.html.twig
+++ b/templates/product/_cart_add_controls.html.twig
@@ -1,4 +1,7 @@
-<turbo-frame id="add-to-cart-controls">
+<turbo-frame
+    id="add-to-cart-controls"
+    data-turbo-form-redirect="true"
+>
 {{ form_start(addToCartForm, {
     attr: { class: 'cart-add-controls d-flex align-items-center justify-content-baseline' }
 }) }}
