diff --git a/templates/product/show.html.twig b/templates/product/show.html.twig
index 7068717..24799b1 100644
--- a/templates/product/show.html.twig
+++ b/templates/product/show.html.twig
@@ -14,7 +14,17 @@
             </div>
         </div>
         <div class="col-8 px-3">
-            <h1>{{ product.name }}</h1>
+            <h1>
+                {{ product.name }}
+                {% if is_granted('ROLE_ADMIN') %}
+                    <a
+                        href="{{ path('product_admin_edit', {
+                            id: product.id
+                        }) }}"
+                        class="btn btn-sm btn-secondary"
+                    >Edit</a>
+                {% endif %}
+            </h1>
             <div>
                 {{ product.description }}
             </div>
