diff --git a/templates/product/show.html.twig b/templates/product/show.html.twig
index 419e0b9d..26c86bc2 100644
--- a/templates/product/show.html.twig
+++ b/templates/product/show.html.twig
@@ -19,7 +19,8 @@
                 {% if is_granted('ROLE_ADMIN') %}
                     <a
                         href="{{ path('product_admin_edit', {
-                            id: product.id
+                            id: product.id,
+                            hash: random()
                         }) }}"
                         class="btn btn-sm btn-secondary"
                         data-turbo-frame="product-info"
