diff --git a/templates/product_admin/edit.html.twig b/templates/product_admin/edit.html.twig
index 4ff3b1b8..65f8be09 100644
--- a/templates/product_admin/edit.html.twig
+++ b/templates/product_admin/edit.html.twig
@@ -5,11 +5,13 @@
 {% block body %}
 <div class="container mt-4">
     <a href="{{ path('product_admin_index') }}"><i class="fas fa-caret-left"></i> Back to list</a>
-    <div class="d-flex justify-content-between">
-        <h1 class="mt-3">Edit Product</h1>
-        {{ include('product_admin/_delete_form.html.twig') }}
-    </div>
+    <turbo-frame id="product-info" target="_top">
+        <div class="d-flex justify-content-between">
+            <h1 class="mt-3">Edit Product</h1>
+            {{ include('product_admin/_delete_form.html.twig') }}
+        </div>
 
-    {{ include('product_admin/_form.html.twig', {'button_label': 'Update'}) }}
+        {{ include('product_admin/_form.html.twig', {'button_label': 'Update'}) }}
+    </turbo-frame>
 </div>
 {% endblock %}
