diff --git a/templates/product/show.html.twig b/templates/product/show.html.twig
index 0a26386..24799b1 100644
--- a/templates/product/show.html.twig
+++ b/templates/product/show.html.twig
@@ -1,8 +1,7 @@
 {% extends 'product/productBase.html.twig' %}
 
 {% block productBody %}
-    <div class="row pt-3 product-show">
-        <turbo-frame id="product-info" target="_top">
+    <turbo-frame id="product-info" target="_top" class="row pt-3 product-show">
         <div class="col-4">
             <img
                 alt="{{ product.name }}"
@@ -42,8 +41,7 @@
                 </div>
             </div>
         </div>
-        </turbo-frame>
-    </div>
+    </turbo-frame>
 
     <hr>
 
