diff --git a/src/Controller/ProductAdminController.php b/src/Controller/ProductAdminController.php
index a32024c1..7e33a08a 100644
--- a/src/Controller/ProductAdminController.php
+++ b/src/Controller/ProductAdminController.php
@@ -83,6 +83,7 @@ class ProductAdminController extends AbstractController
         return $this->renderForm('product_admin/edit.html.twig', [
             'product' => $product,
             'form' => $form,
+            'formTarget' => $request->headers->get('Turbo-Frame', '_top')
         ]);
     }
 
diff --git a/templates/product_admin/_form.html.twig b/templates/product_admin/_form.html.twig
index f8d464bb..cdd16401 100644
--- a/templates/product_admin/_form.html.twig
+++ b/templates/product_admin/_form.html.twig
@@ -1,5 +1,5 @@
 {{ form_start(form, {
-    attr: { 'data-turbo-frame': 'product-info' }
+    attr: { 'data-turbo-frame': formTarget|default('_top') }
 }) }}
     {{ form_widget(form) }}
     <button class="btn btn-primary" formnovalidate>{{ button_label|default('Save') }}</button>
