diff --git a/templates/base.html.twig b/templates/base.html.twig
index e311dd5a..4ce2abb5 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -7,9 +7,12 @@
             <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.5.2/css/bootstrap.min.css" integrity="sha384-JcKb8q3iqJ61gNV9KGb8thSsNjpSL0n8PARn9HuZOnIxN0hoP+VmmDGMN5t9UJ0Z" crossorigin="anonymous">
 
             <link rel="stylesheet" href="{{ asset('styles/app.css') }}">
+            {{ encore_entry_link_tags('app') }}
         {% endblock %}
 
-        {% block javascripts %}{% endblock %}
+        {% block javascripts %}
+            {{ encore_entry_script_tags('app') }}
+        {% endblock %}
     </head>
     <body>
         <div class="page-top">
