diff --git a/src/Controller/ProductAdminController.php b/src/Controller/ProductAdminController.php
index 7266624..b6aa7fa 100644
--- a/src/Controller/ProductAdminController.php
+++ b/src/Controller/ProductAdminController.php
@@ -42,7 +42,9 @@ class ProductAdminController extends AbstractController
             return $this->redirectToRoute('product_admin_index');
         }
 
-        return $this->render('product_admin/new.html.twig', [
+        $template = $request->isXmlHttpRequest() ? '_form.html.twig' : 'new.html.twig';
+
+        return $this->render('product_admin/' . $template, [
             'product' => $product,
             'form' => $form->createView(),
         ]);
