diff --git a/assets/controllers/modal-form_controller.js b/assets/controllers/modal-form_controller.js
index bdbe49e..d51b37e 100644
--- a/assets/controllers/modal-form_controller.js
+++ b/assets/controllers/modal-form_controller.js
@@ -9,6 +9,7 @@ export default class extends Controller {
     }
 
     async openModal(event) {
+        this.modalBodyTarget.innerHTML = 'Loading...';
         const modal = new Modal(this.modalTarget);
         modal.show();
 
