diff --git a/assets/controllers/modal-form_controller.js b/assets/controllers/modal-form_controller.js
index 5fe7ef03..ac2405a2 100644
--- a/assets/controllers/modal-form_controller.js
+++ b/assets/controllers/modal-form_controller.js
@@ -19,7 +19,7 @@ export default class extends Controller {
     async submitForm() {
         const $form = $(this.modalBodyTarget).find('form');
         this.modalBodyTarget.innerHTML = await $.ajax({
-            url: $form.prop('action'),
+            url: this.formUrlValue,
             method: $form.prop('method'),
             data: $form.serialize(),
         });
