diff --git a/assets/controllers/modal-form_controller.js b/assets/controllers/modal-form_controller.js
index d51b37ec1..a92d888bb 100644
--- a/assets/controllers/modal-form_controller.js
+++ b/assets/controllers/modal-form_controller.js
@@ -15,4 +15,9 @@ export default class extends Controller {
 
         this.modalBodyTarget.innerHTML = await $.ajax(this.formUrlValue);
     }
+
+    submitForm() {
+        const $form = $(this.modalBodyTarget).find('form');
+        console.log($form.serialize());
+    }
 }
diff --git a/templates/_modal.html.twig b/templates/_modal.html.twig
index 550a93adf..756851597 100644
--- a/templates/_modal.html.twig
+++ b/templates/_modal.html.twig
@@ -19,7 +19,7 @@
                 <button type="button" class="btn btn-secondary"
                         data-bs-dismiss="modal">Cancel
                 </button>
-                <button type="button" class="btn btn-primary">
+                <button type="button" class="btn btn-primary" data-action="modal-form#submitForm">
                     Save
                 </button>
             </div>
