diff --git a/assets/controllers/modal-form_controller.js b/assets/controllers/modal-form_controller.js
index a92d888..5fe7ef0 100644
--- a/assets/controllers/modal-form_controller.js
+++ b/assets/controllers/modal-form_controller.js
@@ -16,8 +16,12 @@ export default class extends Controller {
         this.modalBodyTarget.innerHTML = await $.ajax(this.formUrlValue);
     }
 
-    submitForm() {
+    async submitForm() {
         const $form = $(this.modalBodyTarget).find('form');
-        console.log($form.serialize());
+        this.modalBodyTarget.innerHTML = await $.ajax({
+            url: $form.prop('action'),
+            method: $form.prop('method'),
+            data: $form.serialize(),
+        });
     }
 }
