diff --git a/assets/controllers/modal-form_controller.js b/assets/controllers/modal-form_controller.js
index 5dcfa4a..87d87f1 100644
--- a/assets/controllers/modal-form_controller.js
+++ b/assets/controllers/modal-form_controller.js
@@ -1,15 +1,17 @@
 import { Controller } from 'stimulus';
 import { Modal } from 'bootstrap';
+import $ from 'jquery';
 
 export default class extends Controller {
-    static targets = ['modal'];
+    static targets = ['modal', 'modalBody'];
     static values = {
         formUrl: String,
     }
 
-    openModal(event) {
-        console.log(this.formUrlValue);
+    async openModal(event) {
         const modal = new Modal(this.modalTarget, {});
         modal.show();
+
+        this.modalBodyTarget.innerHTML = await $.ajax(this.formUrlValue);
     }
 }
diff --git a/templates/_modal.html.twig b/templates/_modal.html.twig
index 6161b48..550a93a 100644
--- a/templates/_modal.html.twig
+++ b/templates/_modal.html.twig
@@ -12,7 +12,7 @@
                         data-bs-dismiss="modal"
                         aria-label="Close"></button>
             </div>
-            <div class="modal-body">
+            <div class="modal-body" data-modal-form-target="modalBody">
                 {{ modalContent|default('Loading...') }}
             </div>
             <div class="modal-footer">
