diff --git a/assets/controllers/submit-confirm_controller.js b/assets/controllers/submit-confirm_controller.js
index 2c60ed1..c8b05ef 100644
--- a/assets/controllers/submit-confirm_controller.js
+++ b/assets/controllers/submit-confirm_controller.js
@@ -1,6 +1,7 @@
 import { Controller } from 'stimulus';
 import Swal from 'sweetalert2';
 
+/* stimulusFetch: 'lazy' */
 export default class extends Controller {
     static values = {
         title: String,
