diff --git a/templates/base.html.twig b/templates/base.html.twig
index 15f533b..32377a0 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -86,6 +86,7 @@
         >
         </div>
 
+        {% block weather_widget %}
         <a class="weatherwidget-io" href="https://forecast7.com/en/40d71n74d01/new-york/" data-label_1="NEW YORK" data-label_2="WEATHER" data-theme="original" >NEW YORK WEATHER</a>
         <script>
         !function (d, s, id) {
@@ -98,6 +99,7 @@
             }
         }(document, 'script', 'weatherwidget-io-js');
         </script>
+        {% endblock %}
 
         <script>
         document.getElementById('logo-img').addEventListener('click', (event) => {
diff --git a/templates/weather/index.html.twig b/templates/weather/index.html.twig
index 99a3a67..f95eb92 100644
--- a/templates/weather/index.html.twig
+++ b/templates/weather/index.html.twig
@@ -16,3 +16,5 @@
     }(document, 'script', 'weatherwidget-io-js');
     </script>
 {% endblock %}
+
+{% block weather_widget %}{% endblock %}
