diff --git a/templates/cart/_featuredSidebar.html.twig b/templates/cart/_featuredSidebar.html.twig
index fd148db9..2d64e39f 100644
--- a/templates/cart/_featuredSidebar.html.twig
+++ b/templates/cart/_featuredSidebar.html.twig
@@ -1,19 +1,21 @@
-<div class="component-light product-show p-3 mb-5">
-    <h5 class="text-center">Featured Product!</h5>
-    <a href="{{ path('app_product', { id: featuredProduct.id }) }}">
-        <img
-            alt="{{ featuredProduct.name }}"
-            src="{{ asset('/uploads/products/'~featuredProduct.imageFilename) }}"
-            class="d-block"
-        >
-    </a>
-    <div class="pt-3">
-        <h6 class="d-flex justify-content-between mb-3">
-            <strong>{{ featuredProduct.name }}</strong>
+<turbo-frame id="cart-sidebar">
+    <div class="component-light product-show p-3 mb-5">
+        <h5 class="text-center">Featured Product!</h5>
+        <a href="{{ path('app_product', { id: featuredProduct.id }) }}">
+            <img
+                alt="{{ featuredProduct.name }}"
+                src="{{ asset('/uploads/products/'~featuredProduct.imageFilename) }}"
+                class="d-block"
+            >
+        </a>
+        <div class="pt-3">
+            <h6 class="d-flex justify-content-between mb-3">
+                <strong>{{ featuredProduct.name }}</strong>
 
-            {{ featuredProduct.priceString|format_currency('USD') }}
-        </h6>
+                {{ featuredProduct.priceString|format_currency('USD') }}
+            </h6>
 
-        {{ include('product/_cart_add_controls.html.twig') }}
+            {{ include('product/_cart_add_controls.html.twig') }}
+        </div>
     </div>
-</div>
+</turbo-frame>
