diff --git a/assets/turbo/turbo-helper.js b/assets/turbo/turbo-helper.js
index d6e0091f..5e07f51b 100644
--- a/assets/turbo/turbo-helper.js
+++ b/assets/turbo/turbo-helper.js
@@ -39,7 +39,9 @@ const TurboHelper = class {
     }
 
     initializeWeatherWidget() {
-        __weatherwidget_init();
+        if (typeof __weatherwidget_init === 'function') {
+            __weatherwidget_init();
+        }
     }
 
     isPreviewRendered() {
