diff --git a/templates/product/productBase.html.twig b/templates/product/productBase.html.twig
index 22302e6b5..f9553e75e 100644
--- a/templates/product/productBase.html.twig
+++ b/templates/product/productBase.html.twig
@@ -6,6 +6,8 @@
         <div class="row">
             <aside class="col-12 col-md-3 order-2 order-md-1">
                 {{ include('product/_categoriesSidebar.html.twig') }}
+
+                <turbo-frame id="weather_widget" src="{{ path('app_weather') }}"></turbo-frame>
             </aside>
 
             <div class="col-12 col-md-9 product-show order-1 order-md-2">
