diff --git a/templates/base.html.twig b/templates/base.html.twig
index 3086274..8c81d8d 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -87,7 +87,7 @@
         </div>
 
         {% block weather_widget %}
-            <turbo-frame id="weather_widget" src="{{ path('app_weather') }}"></turbo-frame>
+            <turbo-frame id="weather_widget" src="{{ path('app_weather') }}" loading="lazy"></turbo-frame>
         {% endblock %}
 
         <script>
