diff --git a/src/Controller/CartController.php b/src/Controller/CartController.php
index 9096277..1c1b443 100644
--- a/src/Controller/CartController.php
+++ b/src/Controller/CartController.php
@@ -21,13 +21,8 @@ class CartController extends AbstractController
     /**
      * @Route("/cart", name="app_cart")
      */
-    public function shoppingCart(CartStorage $cartStorage, ProductRepository  $productRepository): Response
+    public function shoppingCart(CartStorage $cartStorage): Response
     {
-        $featuredProduct = $productRepository->findFeatured();
-        $addToCartForm = $this->createForm(AddItemToCartFormType::class, null, [
-            'product' => $featuredProduct,
-        ]);
-
         return $this->renderForm('cart/cart.html.twig', [
             'cart' => $cartStorage->getOrCreateCart(),
         ]);
