diff --git a/assets/controllers/counter_controller.js b/assets/controllers/counter_controller.js
index bf8e4750e..99dcdf03d 100644
--- a/assets/controllers/counter_controller.js
+++ b/assets/controllers/counter_controller.js
@@ -1,4 +1,5 @@
 import { Controller } from 'stimulus';
+import * as Turbo from '@hotwired/turbo';
 
 export default class extends Controller {
     count = 0;
@@ -9,7 +10,7 @@ export default class extends Controller {
         this.countTarget.innerText = this.count;
 
         if (this.count === 10) {
-            window.location.href = '/you-won';
+            Turbo.visit('/you-won');
         }
     }
 }
