diff --git a/assets/controllers/counter_controller.js b/assets/controllers/counter_controller.js
index e9fd157ad..bf8e4750e 100644
--- a/assets/controllers/counter_controller.js
+++ b/assets/controllers/counter_controller.js
@@ -7,5 +7,9 @@ export default class extends Controller {
     increment() {
         this.count++;
         this.countTarget.innerText = this.count;
+
+        if (this.count === 10) {
+            window.location.href = '/you-won';
+        }
     }
 }
