diff --git a/templates/base.html.twig b/templates/base.html.twig
index 591403a0..f0baa21b 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -93,5 +93,12 @@
             }
         }(document, 'script', 'weatherwidget-io-js');
         </script>
+
+        <script>
+        document.getElementById('logo-img').addEventListener('click', (event) => {
+            event.preventDefault();
+            Turbo.visit('/cart');
+        })
+        </script>
     </body>
 </html>
