diff --git a/assets/controllers/counter_controller.js b/assets/controllers/counter_controller.js
index c90fc046d..3b3f60afb 100644
--- a/assets/controllers/counter_controller.js
+++ b/assets/controllers/counter_controller.js
@@ -1,5 +1,5 @@
 import { Controller } from 'stimulus';
-import { visit } from '@hotwired/turbo';
+import { visit, renderStreamMessage } from '@hotwired/turbo';
 
 export default class extends Controller {
     count = 0;
@@ -9,6 +9,16 @@ export default class extends Controller {
         this.count++;
         this.countTarget.innerText = this.count;
 
+        renderStreamMessage(`
+<turbo-stream action="update" target="flash-container">
+    <template>
+        <div class="alert alert-success">
+            Thanks for clicking ${this.count} times!
+        </div>
+    </template>
+</turbo-stream>
+        `);
+
         if (this.count === 10) {
             visit('/you-won');
         }
