diff --git a/templates/product_admin/edit.html.twig b/templates/product_admin/edit.html.twig
index 83b875790..b01624716 100644
--- a/templates/product_admin/edit.html.twig
+++ b/templates/product_admin/edit.html.twig
@@ -3,6 +3,7 @@
 {% block title %}Edit Product{% endblock %}
 
 {% block body %}
+<div class="container-fluid mt-4">
     <h1>Edit Product</h1>
 
     {{ include('product_admin/_form.html.twig', {'button_label': 'Update'}) }}
@@ -10,4 +11,5 @@
     <a href="{{ path('product_admin_index') }}">back to list</a>
 
     {{ include('product_admin/_delete_form.html.twig') }}
+</div>
 {% endblock %}
diff --git a/templates/product_admin/index.html.twig b/templates/product_admin/index.html.twig
index 87a7a1028..edee43484 100644
--- a/templates/product_admin/index.html.twig
+++ b/templates/product_admin/index.html.twig
@@ -3,6 +3,7 @@
 {% block title %}Product index{% endblock %}
 
 {% block body %}
+<div class="container-fluid mt-4">
     <h1>Product index</h1>
 
     <table class="table">
@@ -44,4 +45,5 @@
     </table>
 
     <a href="{{ path('product_admin_new') }}">Create new</a>
+</div>
 {% endblock %}
diff --git a/templates/product_admin/new.html.twig b/templates/product_admin/new.html.twig
index 1560ef70f..4e2de8cab 100644
--- a/templates/product_admin/new.html.twig
+++ b/templates/product_admin/new.html.twig
@@ -3,9 +3,11 @@
 {% block title %}New Product{% endblock %}
 
 {% block body %}
+<div class="container-fluid mt-4">
     <h1>Create new Product</h1>
 
     {{ include('product_admin/_form.html.twig') }}
 
     <a href="{{ path('product_admin_index') }}">back to list</a>
+</div>
 {% endblock %}
diff --git a/templates/product_admin/show.html.twig b/templates/product_admin/show.html.twig
index 4bb4073d2..79dd56ff3 100644
--- a/templates/product_admin/show.html.twig
+++ b/templates/product_admin/show.html.twig
@@ -3,6 +3,7 @@
 {% block title %}Product{% endblock %}
 
 {% block body %}
+<div class="container-fluid mt-4">
     <h1>Product</h1>
 
     <table class="table">
@@ -47,4 +48,5 @@
     <a href="{{ path('product_admin_edit', {'id': product.id}) }}">edit</a>
 
     {{ include('product_admin/_delete_form.html.twig') }}
+</div>
 {% endblock %}
