diff --git a/templates/_modal.html.twig b/templates/_modal.html.twig
index df87b80c..da621b5d 100644
--- a/templates/_modal.html.twig
+++ b/templates/_modal.html.twig
@@ -15,6 +15,7 @@
             <turbo-frame
                 class="modal-body"
                 src="{{ modalSrc }}"
+                id="{{ id }}"
             >
                 {{ modalContent|default('Loading...') }}
             </turbo-frame>
diff --git a/templates/product_admin/index.html.twig b/templates/product_admin/index.html.twig
index b1ff5b16..93766ec0 100644
--- a/templates/product_admin/index.html.twig
+++ b/templates/product_admin/index.html.twig
@@ -26,6 +26,7 @@
             {{ include('_modal.html.twig', {
                 modalTitle: 'Add a new Product',
                 modalSrc: path('product_admin_new'),
+                id: 'product-info',
             }) }}
         </div>
     </div>
