diff --git a/assets/controllers/modal-form_controller.js b/assets/controllers/modal-form_controller.js
index e6113c0..7a7a523 100644
--- a/assets/controllers/modal-form_controller.js
+++ b/assets/controllers/modal-form_controller.js
@@ -8,8 +8,14 @@ export default class extends Controller {
     connect() {
         document.addEventListener('turbo:before-fetch-response', (event) => {
             console.log(event);
-            if (event.detail.success) {
-                //this.modal.hide();
+            if (!this.modal || !this.modal._isShown) {
+                return;
+            }
+
+            const fetchResponse = event.detail.fetchResponse;
+            if (fetchResponse.succeeded && fetchResponse.redirected) {
+                event.preventDefault();
+                this.modal.hide();
             }
         });
     }
