diff --git a/src/Controller/ProductAdminController.php b/src/Controller/ProductAdminController.php
index 174aa7375..09bc05225 100644
--- a/src/Controller/ProductAdminController.php
+++ b/src/Controller/ProductAdminController.php
@@ -11,7 +11,7 @@ use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
 
 /**
- * @Route("/product/admin")
+ * @Route("/admin/product")
  */
 class ProductAdminController extends AbstractController
 {
