diff --git a/assets/controllers/modal-form_controller.js b/assets/controllers/modal-form_controller.js
index 5fed7fad..292ae5f0 100644
--- a/assets/controllers/modal-form_controller.js
+++ b/assets/controllers/modal-form_controller.js
@@ -29,7 +29,6 @@ export default class extends Controller {
         if (fetchResponse.succeeded && fetchResponse.redirected) {
             event.preventDefault();
             Turbo.visit(fetchResponse.location);
-            this.modal.hide();
         }
     }
 }
