diff --git a/src/Controller/ProductAdminController.php b/src/Controller/ProductAdminController.php
index 09bc05225..726662477 100644
--- a/src/Controller/ProductAdminController.php
+++ b/src/Controller/ProductAdminController.php
@@ -21,7 +21,7 @@ class ProductAdminController extends AbstractController
     public function index(ProductRepository $productRepository): Response
     {
         return $this->render('product_admin/index.html.twig', [
-            'products' => $productRepository->findAll(),
+            'products' => $productRepository->findBy([], ['id' => 'DESC']),
         ]);
     }
 
