diff --git a/src/Form/ProductType.php b/src/Form/ProductType.php
index b0f2b5169..dfaec8d49 100644
--- a/src/Form/ProductType.php
+++ b/src/Form/ProductType.php
@@ -14,13 +14,8 @@ class ProductType extends AbstractType
         $builder
             ->add('name')
             ->add('description')
-            ->add('brand')
-            ->add('weight')
             ->add('price')
-            ->add('stockQuantity')
-            ->add('imageFilename')
             ->add('category')
-            ->add('colors')
         ;
     }
 
