diff --git a/assets/controllers/modal-form_controller.js b/assets/controllers/modal-form_controller.js
index 96b3033..432f2e7 100644
--- a/assets/controllers/modal-form_controller.js
+++ b/assets/controllers/modal-form_controller.js
@@ -8,6 +8,9 @@ export default class extends Controller {
     connect() {
         this.element.addEventListener('turbo:submit-end', (event) => {
             console.log(event);
+            if (event.detail.success) {
+                this.modal.hide();
+            }
         });
     }
 
