diff --git a/src/Controller/ProductAdminController.php b/src/Controller/ProductAdminController.php
new file mode 100644
index 000000000..174aa7375
--- /dev/null
+++ b/src/Controller/ProductAdminController.php
@@ -0,0 +1,94 @@
+<?php
+
+namespace App\Controller;
+
+use App\Entity\Product;
+use App\Form\ProductType;
+use App\Repository\ProductRepository;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @Route("/product/admin")
+ */
+class ProductAdminController extends AbstractController
+{
+    /**
+     * @Route("/", name="product_admin_index", methods={"GET"})
+     */
+    public function index(ProductRepository $productRepository): Response
+    {
+        return $this->render('product_admin/index.html.twig', [
+            'products' => $productRepository->findAll(),
+        ]);
+    }
+
+    /**
+     * @Route("/new", name="product_admin_new", methods={"GET","POST"})
+     */
+    public function new(Request $request): Response
+    {
+        $product = new Product();
+        $form = $this->createForm(ProductType::class, $product);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $entityManager = $this->getDoctrine()->getManager();
+            $entityManager->persist($product);
+            $entityManager->flush();
+
+            return $this->redirectToRoute('product_admin_index');
+        }
+
+        return $this->render('product_admin/new.html.twig', [
+            'product' => $product,
+            'form' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * @Route("/{id}", name="product_admin_show", methods={"GET"})
+     */
+    public function show(Product $product): Response
+    {
+        return $this->render('product_admin/show.html.twig', [
+            'product' => $product,
+        ]);
+    }
+
+    /**
+     * @Route("/{id}/edit", name="product_admin_edit", methods={"GET","POST"})
+     */
+    public function edit(Request $request, Product $product): Response
+    {
+        $form = $this->createForm(ProductType::class, $product);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $this->getDoctrine()->getManager()->flush();
+
+            return $this->redirectToRoute('product_admin_index');
+        }
+
+        return $this->render('product_admin/edit.html.twig', [
+            'product' => $product,
+            'form' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * @Route("/{id}", name="product_admin_delete", methods={"POST"})
+     */
+    public function delete(Request $request, Product $product): Response
+    {
+        if ($this->isCsrfTokenValid('delete'.$product->getId(), $request->request->get('_token'))) {
+            $entityManager = $this->getDoctrine()->getManager();
+            $entityManager->remove($product);
+            $entityManager->flush();
+        }
+
+        return $this->redirectToRoute('product_admin_index');
+    }
+}
diff --git a/src/Form/ProductType.php b/src/Form/ProductType.php
new file mode 100644
index 000000000..b0f2b5169
--- /dev/null
+++ b/src/Form/ProductType.php
@@ -0,0 +1,33 @@
+<?php
+
+namespace App\Form;
+
+use App\Entity\Product;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class ProductType extends AbstractType
+{
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('name')
+            ->add('description')
+            ->add('brand')
+            ->add('weight')
+            ->add('price')
+            ->add('stockQuantity')
+            ->add('imageFilename')
+            ->add('category')
+            ->add('colors')
+        ;
+    }
+
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver->setDefaults([
+            'data_class' => Product::class,
+        ]);
+    }
+}
diff --git a/templates/product_admin/_delete_form.html.twig b/templates/product_admin/_delete_form.html.twig
new file mode 100644
index 000000000..f1344bd13
--- /dev/null
+++ b/templates/product_admin/_delete_form.html.twig
@@ -0,0 +1,4 @@
+<form method="post" action="{{ path('product_admin_delete', {'id': product.id}) }}" onsubmit="return confirm('Are you sure you want to delete this item?');">
+    <input type="hidden" name="_token" value="{{ csrf_token('delete' ~ product.id) }}">
+    <button class="btn">Delete</button>
+</form>
diff --git a/templates/product_admin/_form.html.twig b/templates/product_admin/_form.html.twig
new file mode 100644
index 000000000..bf20b98fb
--- /dev/null
+++ b/templates/product_admin/_form.html.twig
@@ -0,0 +1,4 @@
+{{ form_start(form) }}
+    {{ form_widget(form) }}
+    <button class="btn">{{ button_label|default('Save') }}</button>
+{{ form_end(form) }}
diff --git a/templates/product_admin/edit.html.twig b/templates/product_admin/edit.html.twig
new file mode 100644
index 000000000..83b875790
--- /dev/null
+++ b/templates/product_admin/edit.html.twig
@@ -0,0 +1,13 @@
+{% extends 'base.html.twig' %}
+
+{% block title %}Edit Product{% endblock %}
+
+{% block body %}
+    <h1>Edit Product</h1>
+
+    {{ include('product_admin/_form.html.twig', {'button_label': 'Update'}) }}
+
+    <a href="{{ path('product_admin_index') }}">back to list</a>
+
+    {{ include('product_admin/_delete_form.html.twig') }}
+{% endblock %}
diff --git a/templates/product_admin/index.html.twig b/templates/product_admin/index.html.twig
new file mode 100644
index 000000000..87a7a1028
--- /dev/null
+++ b/templates/product_admin/index.html.twig
@@ -0,0 +1,47 @@
+{% extends 'base.html.twig' %}
+
+{% block title %}Product index{% endblock %}
+
+{% block body %}
+    <h1>Product index</h1>
+
+    <table class="table">
+        <thead>
+            <tr>
+                <th>Id</th>
+                <th>Name</th>
+                <th>Description</th>
+                <th>Brand</th>
+                <th>Weight</th>
+                <th>Price</th>
+                <th>StockQuantity</th>
+                <th>ImageFilename</th>
+                <th>actions</th>
+            </tr>
+        </thead>
+        <tbody>
+        {% for product in products %}
+            <tr>
+                <td>{{ product.id }}</td>
+                <td>{{ product.name }}</td>
+                <td>{{ product.description }}</td>
+                <td>{{ product.brand }}</td>
+                <td>{{ product.weight }}</td>
+                <td>{{ product.price }}</td>
+                <td>{{ product.stockQuantity }}</td>
+                <td>{{ product.imageFilename }}</td>
+                <td>
+                    <a href="{{ path('product_admin_show', {'id': product.id}) }}">show</a>
+                    <a href="{{ path('product_admin_edit', {'id': product.id}) }}">edit</a>
+                </td>
+            </tr>
+        {% else %}
+            <tr>
+                <td colspan="9">no records found</td>
+            </tr>
+        {% endfor %}
+        </tbody>
+    </table>
+
+    <a href="{{ path('product_admin_new') }}">Create new</a>
+{% endblock %}
diff --git a/templates/product_admin/new.html.twig b/templates/product_admin/new.html.twig
new file mode 100644
index 000000000..1560ef70f
--- /dev/null
+++ b/templates/product_admin/new.html.twig
@@ -0,0 +1,11 @@
+{% extends 'base.html.twig' %}
+
+{% block title %}New Product{% endblock %}
+
+{% block body %}
+    <h1>Create new Product</h1>
+
+    {{ include('product_admin/_form.html.twig') }}
+
+    <a href="{{ path('product_admin_index') }}">back to list</a>
+{% endblock %}
diff --git a/templates/product_admin/show.html.twig b/templates/product_admin/show.html.twig
new file mode 100644
index 000000000..4bb4073d2
--- /dev/null
+++ b/templates/product_admin/show.html.twig
@@ -0,0 +1,50 @@
+{% extends 'base.html.twig' %}
+
+{% block title %}Product{% endblock %}
+
+{% block body %}
+    <h1>Product</h1>
+
+    <table class="table">
+        <tbody>
+            <tr>
+                <th>Id</th>
+                <td>{{ product.id }}</td>
+            </tr>
+            <tr>
+                <th>Name</th>
+                <td>{{ product.name }}</td>
+            </tr>
+            <tr>
+                <th>Description</th>
+                <td>{{ product.description }}</td>
+            </tr>
+            <tr>
+                <th>Brand</th>
+                <td>{{ product.brand }}</td>
+            </tr>
+            <tr>
+                <th>Weight</th>
+                <td>{{ product.weight }}</td>
+            </tr>
+            <tr>
+                <th>Price</th>
+                <td>{{ product.price }}</td>
+            </tr>
+            <tr>
+                <th>StockQuantity</th>
+                <td>{{ product.stockQuantity }}</td>
+            </tr>
+            <tr>
+                <th>ImageFilename</th>
+                <td>{{ product.imageFilename }}</td>
+            </tr>
+        </tbody>
+    </table>
+
+    <a href="{{ path('product_admin_index') }}">back to list</a>
+
+    <a href="{{ path('product_admin_edit', {'id': product.id}) }}">edit</a>
+
+    {{ include('product_admin/_delete_form.html.twig') }}
+{% endblock %}
