diff --git a/assets/controllers/modal-form_controller.js b/assets/controllers/modal-form_controller.js
index ad3b7492..96b30339 100644
--- a/assets/controllers/modal-form_controller.js
+++ b/assets/controllers/modal-form_controller.js
@@ -5,6 +5,12 @@ export default class extends Controller {
     static targets = ['modal'];
     modal = null;
 
+    connect() {
+        this.element.addEventListener('turbo:submit-end', (event) => {
+            console.log(event);
+        });
+    }
+
     async openModal(event) {
         this.modal = new Modal(this.modalTarget);
         this.modal.show();
