diff --git a/templates/_modal.html.twig b/templates/_modal.html.twig
index 41aa07ac..f907c361 100644
--- a/templates/_modal.html.twig
+++ b/templates/_modal.html.twig
@@ -16,6 +16,7 @@
                 class="modal-body"
                 src="{{ modalSrc }}"
                 id="{{ id }}"
+                loading="lazy"
             >
                 {{ modalContent|default('Loading...') }}
             </turbo-frame>
