diff --git a/assets/controllers/modal-form_controller.js b/assets/controllers/modal-form_controller.js
index 9d5e4839d..0ff9f8449 100644
--- a/assets/controllers/modal-form_controller.js
+++ b/assets/controllers/modal-form_controller.js
@@ -7,7 +7,8 @@ export default class extends Controller {
     modal = null;
 
     connect() {
-        document.addEventListener('turbo:before-fetch-response', this.beforeFetchResponse);
+        this.boundBeforeFetchResponse = this.beforeFetchResponse.bind(this);
+        document.addEventListener('turbo:before-fetch-response', this.boundBeforeFetchResponse);
     }
 
     async openModal(event) {
