diff --git a/assets/controllers/modal-form_controller.js b/assets/controllers/modal-form_controller.js
index 1fe0a70..5dcfa4a 100644
--- a/assets/controllers/modal-form_controller.js
+++ b/assets/controllers/modal-form_controller.js
@@ -3,8 +3,12 @@ import { Modal } from 'bootstrap';
 
 export default class extends Controller {
     static targets = ['modal'];
+    static values = {
+        formUrl: String,
+    }
 
     openModal(event) {
+        console.log(this.formUrlValue);
         const modal = new Modal(this.modalTarget, {});
         modal.show();
     }
diff --git a/templates/product_admin/index.html.twig b/templates/product_admin/index.html.twig
index 9e6b821..93d133f 100644
--- a/templates/product_admin/index.html.twig
+++ b/templates/product_admin/index.html.twig
@@ -7,7 +7,9 @@
     <div class="d-flex flex-row">
         <h1 class="me-3">Product index</h1>
 
-        <div {{ stimulus_controller('modal-form') }}>
+        <div {{ stimulus_controller('modal-form', {
+            formUrl: path('product_admin_new')
+        }) }}>
             <button
                 class="btn btn-primary btn-sm"
                 data-action="modal-form#openModal"
