diff --git a/assets/controllers/modal-form_controller.js b/assets/controllers/modal-form_controller.js
index 54fe707..9d5e483 100644
--- a/assets/controllers/modal-form_controller.js
+++ b/assets/controllers/modal-form_controller.js
@@ -7,21 +7,23 @@ export default class extends Controller {
     modal = null;
 
     connect() {
-        document.addEventListener('turbo:before-fetch-response', (event) => {
-            if (!this.modal || !this.modal._isShown) {
-                return;
-            }
-
-            const fetchResponse = event.detail.fetchResponse;
-            if (fetchResponse.succeeded && fetchResponse.redirected) {
-                event.preventDefault();
-                Turbo.visit(fetchResponse.location);
-            }
-        });
+        document.addEventListener('turbo:before-fetch-response', this.beforeFetchResponse);
     }
 
     async openModal(event) {
         this.modal = new Modal(this.modalTarget);
         this.modal.show();
     }
+
+    beforeFetchResponse(event) {
+        if (!this.modal || !this.modal._isShown) {
+            return;
+        }
+
+        const fetchResponse = event.detail.fetchResponse;
+        if (fetchResponse.succeeded && fetchResponse.redirected) {
+            event.preventDefault();
+            Turbo.visit(fetchResponse.location);
+        }
+    }
 }
