diff --git a/assets/controllers/color-square_controller.js b/assets/controllers/color-square_controller.js
index 1cf9d93..a089914 100644
--- a/assets/controllers/color-square_controller.js
+++ b/assets/controllers/color-square_controller.js
@@ -10,10 +10,12 @@ export default class extends Controller {
     }
 
     selectColor(event) {
-        const newColorId = event.currentTarget.dataset.colorId;
+        this.setSelectedColor(event.currentTarget.dataset.colorId)
+    }
 
+    setSelectedColor(newColorId) {
         if (newColorId === this.selectedColorId) {
-            event.currentTarget.classList.remove('selected');
+            this.findSelectedColorSquare().classList.remove('selected');
 
             this.selectedColorId = null;
             this.selectTarget.value = '';
@@ -27,7 +29,14 @@ export default class extends Controller {
             element.classList.remove('selected');
         });
 
-        event.currentTarget.classList.add('selected');
+        this.findSelectedColorSquare().classList.add('selected');
         this.selectTarget.value = this.selectedColorId;
     }
+
+    /**
+     * @return {Element|null}
+     */
+    findSelectedColorSquare() {
+        return this.colorSquareTargets.find((element) => element.dataset.colorId === this.selectedColorId);
+    }
 }
