diff --git a/tutorial/FeaturedProduct.js b/tutorial/FeaturedProduct.js
deleted file mode 100644
index 91b85ac6..00000000
--- a/tutorial/FeaturedProduct.js
+++ /dev/null
@@ -1,25 +0,0 @@
-import React from 'react';
-
-export default function(props) {
-    return (
-        <div>
-            <div className="component-light product-show p-3 mb-5">
-                <h5 className="text-center">Featured Product!</h5>
-                <a href={'/product/'+props.product.id}>
-                    <img
-                        alt={ props.product.name }
-                        src={props.product.imageUrl}
-                        className="d-block"
-                    />
-                </a>
-                <div className="pt-3">
-                    <h6>
-                        {props.product.name}
-                    </h6>
-
-                    <a href={'/product/'+props.product.id} className="btn btn-sm btn-primary">More info</a>
-                </div>
-            </div>
-        </div>
-    )
-}
diff --git a/tutorial/MadeWithLove.js b/tutorial/MadeWithLove.js
deleted file mode 100644
index 7507a9d8..00000000
--- a/tutorial/MadeWithLove.js
+++ /dev/null
@@ -1,30 +0,0 @@
-import React from 'react';
-
-export default class MadeWithLove extends React.Component {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            hearts: 1,
-        };
-    }
-
-    render() {
-        return (
-            <div>
-                <span>Built with </span>
-                <span
-                    onClick={() => this.setState({
-                        hearts: this.state.hearts + 1,
-                    })}
-                >
-                    {'❤️'.repeat(this.state.hearts)}
-                </span>
-                <span> by your friends at </span>
-                <a href="https://symfonycasts.com" target="_blank">
-                    SymfonyCasts
-                </a>
-            </div>
-        )
-    }
-}
