diff --git a/src/Controller/CartController.php b/src/Controller/CartController.php
index ae46e1839..13745e2e3 100644
--- a/src/Controller/CartController.php
+++ b/src/Controller/CartController.php
@@ -46,7 +46,7 @@ class CartController extends AbstractController
     }
 
     /**
-     * @Route("/product/{id}", name="app_cart_add_item", methods={"POST"})
+     * @Route("/product/{id}/cart", name="app_cart_add_item", methods={"POST"})
      */
     public function addItemToCart(Product $product, Request $request, CategoryRepository $categoryRepository, CartStorage $cartStorage)
     {
