diff --git a/src/Controller/ProductController.php b/src/Controller/ProductController.php
index 484d46a..91aecaf 100644
--- a/src/Controller/ProductController.php
+++ b/src/Controller/ProductController.php
@@ -56,4 +56,12 @@ class ProductController extends AbstractController
             'addToCartForm' => $addToCartForm->createView()
         ]);
     }
+
+    /**
+     * @Route("/i-like-counting")
+     */
+    public function iLikeCounting()
+    {
+        return $this->render('product/counting.html.twig');
+    }
 }
diff --git a/templates/product/counting.html.twig b/templates/product/counting.html.twig
new file mode 100644
index 0000000..b81a8b8
--- /dev/null
+++ b/templates/product/counting.html.twig
@@ -0,0 +1,11 @@
+{% extends 'base.html.twig' %}
+
+{% block body %}
+<div class="container-fluid mt-4">
+    <div class="row">
+        <div class="col-12">
+            Wow! You really like counting! Me too!
+        </div>
+    </div>
+</div>
+{% endblock %}
