diff --git a/src/Entity/Product.php b/src/Entity/Product.php
index c160e5496..ca28409ec 100644
--- a/src/Entity/Product.php
+++ b/src/Entity/Product.php
@@ -98,7 +98,7 @@ class Product
         return $this->name;
     }
 
-    public function setName(string $name): self
+    public function setName(?string $name): self
     {
         $this->name = $name;
 
@@ -122,7 +122,7 @@ class Product
         return $this->brand;
     }
 
-    public function setBrand(string $brand): self
+    public function setBrand(?string $brand): self
     {
         $this->brand = $brand;
 
diff --git a/src/Entity/Review.php b/src/Entity/Review.php
index 418b4a09c..f41106219 100644
--- a/src/Entity/Review.php
+++ b/src/Entity/Review.php
@@ -70,7 +70,7 @@ class Review
         return substr($this->content, 0, 50).'...';
     }
 
-    public function setContent(string $content): self
+    public function setContent(?string $content): self
     {
         $this->content = $content;
 
@@ -92,7 +92,7 @@ class Review
         return $this->stars;
     }
 
-    public function setStars(int $stars): self
+    public function setStars(?int $stars): self
     {
         $this->stars = $stars;
 
