diff --git a/assets/util/add-transition.js b/assets/util/add-transition.js
index 698d67845..b0ed4f3ff 100644
--- a/assets/util/add-transition.js
+++ b/assets/util/add-transition.js
@@ -1,6 +1,6 @@
 import { useTransition } from 'stimulus-use';
 
-export function addFadeTransition(controller, element) {
+export function addFadeTransition(controller, element, options) {
     useTransition(controller, {
         element,
         enterActive: 'fade-enter-active',
@@ -10,5 +10,6 @@ export function addFadeTransition(controller, element) {
         leaveFrom: 'fade-leave-from',
         leaveTo: 'fade-leave-to',
         hiddenClass: 'd-none',
+        ...options
     });
 }
