diff --git a/templates/base.html.twig b/templates/base.html.twig
index f0baa21b..50881ffb 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -10,6 +10,8 @@
 
         {% block javascripts %}
             {{ encore_entry_script_tags('app') }}
+
+            <script src="//instant.page/5.1.0" type="module" integrity="sha384-by67kQnR+pyfy8yWP4kPO12fHKRLHZPfEsiSXR8u2IKcTdxD805MGUXBzVPnkLHw"></script>
         {% endblock %}
     </head>
     <body>
