diff --git a/assets/controllers/made-with-love_controller.js b/assets/controllers/made-with-love_controller.js
new file mode 100644
index 00000000..ca0d704e
--- /dev/null
+++ b/assets/controllers/made-with-love_controller.js
@@ -0,0 +1,7 @@
+import { Controller } from 'stimulus';
+
+export default class extends Controller {
+    connect() {
+        console.log('❤️');
+    }
+}
diff --git a/templates/base.html.twig b/templates/base.html.twig
index cbfb9749..0e069350 100644
--- a/templates/base.html.twig
+++ b/templates/base.html.twig
@@ -62,11 +62,10 @@
 
         {% block body %}{% endblock %}
 
-        <div class="footer mb-0">
-            Built with<span> &lt;3 </span>by your friends at
-            <a href="https://symfonycasts.com" target="_blank">
-                SymfonyCasts
-            </a>
+        <div
+            class="footer mb-0"
+            {{ stimulus_controller('made-with-love') }}
+        >
         </div>
     </body>
 </html>
