diff --git a/assets/controllers/featured-product-react_controller.js b/assets/controllers/featured-product-react_controller.js
index 6c67e67..dbe3a63 100644
--- a/assets/controllers/featured-product-react_controller.js
+++ b/assets/controllers/featured-product-react_controller.js
@@ -4,9 +4,13 @@ import React from 'react';
 import FeaturedProduct from '../components/FeaturedProduct';
 
 export default class extends Controller {
+    static values = {
+        product: Object
+    }
+
     connect() {
         ReactDOM.render(
-            <FeaturedProduct />,
+            <FeaturedProduct product={this.productValue} />,
             this.element
         )
     }
