diff --git a/assets/controllers/cart-list_controller.js b/assets/controllers/cart-list_controller.js
deleted file mode 100644
index 04607159..00000000
--- a/assets/controllers/cart-list_controller.js
+++ /dev/null
@@ -1,14 +0,0 @@
-import { Controller } from 'stimulus';
-
-export default class extends Controller {
-    static values = {
-        cartRefreshUrl: String,
-    }
-
-    async removeItem(event) {
-        event.currentTarget.classList.add('removing');
-
-        const response = await fetch(this.cartRefreshUrlValue);
-        this.element.innerHTML = await response.text();
-    }
-}
diff --git a/templates/cart/_cartList.html.twig b/templates/cart/_cartList.html.twig
index 5da81fc5..f512bec7 100644
--- a/templates/cart/_cartList.html.twig
+++ b/templates/cart/_cartList.html.twig
@@ -14,7 +14,7 @@
     {% for item in cart.items %}
         <div
             class="cart-item row p-3"
-            data-action="submit-confirm:async:submitted->cart-list#removeItem"
+            data-action="submit-confirm:async:submitted->reload-content#refreshContent"
         >
             <div class="col-2">
                 {{ item.product.name }}
diff --git a/templates/cart/cart.html.twig b/templates/cart/cart.html.twig
index f33e6209..2e243996 100644
--- a/templates/cart/cart.html.twig
+++ b/templates/cart/cart.html.twig
@@ -13,8 +13,8 @@
                 <h1>Shopping Cart</h1>
                 <div
                     class="component-light p-3"
-                    {{ stimulus_controller('cart-list', {
-                        cartRefreshUrl: path('_app_cart_list')
+                    {{ stimulus_controller('reload-content', {
+                        url: path('_app_cart_list')
                     }) }}
                 >
                     {{ include('cart/_cartList.html.twig') }}
