diff --git a/assets/controllers/reload-content_controller.js b/assets/controllers/reload-content_controller.js
index 1b587c8e..caba9c5e 100644
--- a/assets/controllers/reload-content_controller.js
+++ b/assets/controllers/reload-content_controller.js
@@ -7,9 +7,11 @@ export default class extends Controller {
     }
 
     async refreshContent(event) {
-        this.contentTarget.style.opacity = .5;
+        const target = this.hasContentTarget ? this.contentTarget : this.element;
+
+        target.style.opacity = .5;
         const response = await fetch(this.urlValue);
-        this.contentTarget.innerHTML = await response.text();
-        this.contentTarget.style.opacity = 1;
+        target.innerHTML = await response.text();
+        target.style.opacity = 1;
     }
 }
