diff --git a/assets/controllers/reload-content_controller.js b/assets/controllers/reload-content_controller.js
index 337c387..1b587c8 100644
--- a/assets/controllers/reload-content_controller.js
+++ b/assets/controllers/reload-content_controller.js
@@ -7,7 +7,9 @@ export default class extends Controller {
     }
 
     async refreshContent(event) {
+        this.contentTarget.style.opacity = .5;
         const response = await fetch(this.urlValue);
         this.contentTarget.innerHTML = await response.text();
+        this.contentTarget.style.opacity = 1;
     }
 }
