diff --git a/src/Controller/ProductAdminController.php b/src/Controller/ProductAdminController.php
index b46735d..f3ae653 100644
--- a/src/Controller/ProductAdminController.php
+++ b/src/Controller/ProductAdminController.php
@@ -20,7 +20,7 @@ class ProductAdminController extends AbstractController
      */
     public function index(ProductRepository $productRepository, Request $request): Response
     {
-        $template = $request->isXmlHttpRequest() ? '_list.html.twig' : 'index.html.twig';
+        $template = $request->query->get('ajax') ? '_list.html.twig' : 'index.html.twig';
 
         return $this->render('product_admin/' . $template, [
             'products' => $productRepository->findBy([], ['id' => 'DESC']),
diff --git a/templates/product_admin/index.html.twig b/templates/product_admin/index.html.twig
index beae12f..5f4686f 100644
--- a/templates/product_admin/index.html.twig
+++ b/templates/product_admin/index.html.twig
@@ -6,7 +6,7 @@
 <div
     class="container-fluid mt-4"
     {{ stimulus_controller('reload-content', {
-        url: path('product_admin_index')
+        url: path('product_admin_index', { ajax: 1 })
     }) }}
     data-action="modal-form:success->reload-content#refreshContent"
 >
