diff --git a/assets/controllers/modal-form_controller.js b/assets/controllers/modal-form_controller.js
index df510cd4f..b2c1b1c76 100644
--- a/assets/controllers/modal-form_controller.js
+++ b/assets/controllers/modal-form_controller.js
@@ -11,7 +11,7 @@ export default class extends Controller {
     modal = null;
 
     connect() {
-        useDispatch(this, {debug: true});
+        useDispatch(this);
     }
 
     async openModal(event) {
diff --git a/templates/product_admin/index.html.twig b/templates/product_admin/index.html.twig
index 9a857f438..beae12feb 100644
--- a/templates/product_admin/index.html.twig
+++ b/templates/product_admin/index.html.twig
@@ -8,6 +8,7 @@
     {{ stimulus_controller('reload-content', {
         url: path('product_admin_index')
     }) }}
+    data-action="modal-form:success->reload-content#refreshContent"
 >
     <div class="d-flex flex-row">
         <h1 class="me-3">Product index</h1>
