diff --git a/assets/turbo/turbo-helper.js b/assets/turbo/turbo-helper.js
index 3e14bba..bd82bfd 100644
--- a/assets/turbo/turbo-helper.js
+++ b/assets/turbo/turbo-helper.js
@@ -101,7 +101,7 @@ const TurboHelper = class {
         });
     }
 
-    async beforeFetchResponse(event) {
+    beforeFetchResponse(event) {
         const fetchResponse = event.detail.fetchResponse;
         if (!fetchResponse.succeeded || !fetchResponse.redirected) {
             return;
@@ -113,12 +113,8 @@ const TurboHelper = class {
 
         event.preventDefault();
         Turbo.clearCache();
-        const snapshot = Turbo.PageSnapshot.fromHTMLString(await fetchResponse.responseHTML);
-        Turbo.navigator.view.snapshotCache.put(fetchResponse.location, snapshot)
 
-        Turbo.visit(fetchResponse.location, {
-            action: 'restore'
-        });
+        Turbo.visit(fetchResponse.location);
     }
 
     getCurrentFrame() {
