diff --git a/src/Controller/ProductController.php b/src/Controller/ProductController.php
index 5eec9e1..7431f91 100644
--- a/src/Controller/ProductController.php
+++ b/src/Controller/ProductController.php
@@ -86,7 +86,7 @@ class ProductController extends AbstractController
                 $entityManager->persist($reviewForm->getData());
                 $entityManager->flush();
 
-                $this->addFlash('success', 'Thanks for your review! I like you!');
+                $this->addFlash('review_success', 'Thanks for your review! I like you!');
 
                 return $this->redirectToRoute('app_product_reviews', [
                     'id' => $product->getId(),
diff --git a/templates/product/_reviews.html.twig b/templates/product/_reviews.html.twig
index 2cd343d..ad6911f 100644
--- a/templates/product/_reviews.html.twig
+++ b/templates/product/_reviews.html.twig
@@ -12,6 +12,10 @@
 
 <hr>
 
+{% for flash in app.flashes('review_success') %}
+    <div class="alert alert-success">{{ flash }}</div>
+{% endfor %}
+
 {% if reviewForm|default(false) %}
     <h4>Post your own review</h4>
     {{ form_start(reviewForm, {
