diff --git a/templates/product/_reviews.html.twig b/templates/product/_reviews.html.twig
index 2a3093fb..2cd343df 100644
--- a/templates/product/_reviews.html.twig
+++ b/templates/product/_reviews.html.twig
@@ -1,3 +1,4 @@
+<turbo-frame id="product-review">
 {% for review in product.reviews %}
     <div class="component-light my-3 p-3">
         <p><i class="fas fa-user-circle me-2"></i>{{ review.owner.email }} <i class="fas fa-star ms-4"></i> {{ review.stars }}/5</p>
@@ -24,3 +25,4 @@
 {% elseif not is_granted('ROLE_USER') %}
     <p><a href="{{ path('app_login') }}">Log in</a> to post your review</p>
 {% endif %}
+</turbo-frame>
