diff --git a/assets/turbo/turbo-helper.js b/assets/turbo/turbo-helper.js
index 97324d46..b668cb63 100644
--- a/assets/turbo/turbo-helper.js
+++ b/assets/turbo/turbo-helper.js
@@ -5,10 +5,13 @@ const TurboHelper = class {
         document.addEventListener('turbo:before-cache', () => {
             this.closeModal();
             this.closeSweetalert();
+            this.reenableSubmitButtons();
         });
 
         document.addEventListener('turbo:submit-start', (event) => {
-            event.detail.formSubmission.submitter.toggleAttribute('disabled', true);
+            const submitter = event.detail.formSubmission.submitter;
+            submitter.toggleAttribute('disabled', true);
+            submitter.classList.add('turbo-submit-disabled');
         })
 
         this.initializeTransitions();
@@ -85,6 +88,13 @@ const TurboHelper = class {
             }
         });
     }
+
+    reenableSubmitButtons() {
+        document.querySelectorAll('.turbo-submit-disabled').forEach((button) => {
+            button.toggleAttribute('disabled', false);
+            button.classList.remove('turbo-submit-disabled');
+        });
+    }
 }
 
 export default new TurboHelper();
