diff --git a/assets/turbo/turbo-helper.js b/assets/turbo/turbo-helper.js
index c79731a..02e51e0 100644
--- a/assets/turbo/turbo-helper.js
+++ b/assets/turbo/turbo-helper.js
@@ -8,7 +8,7 @@ const TurboHelper = class {
         });
 
         document.addEventListener('turbo:submit-start', (event) => {
-            console.log('submit-start', event);
+            event.detail.formSubmission.submitter.toggleAttribute('disabled', true);
         })
 
         this.initializeTransitions();
