diff --git a/templates/product/_reviews.html.twig b/templates/product/_reviews.html.twig
index ad6911f54..7c402287c 100644
--- a/templates/product/_reviews.html.twig
+++ b/templates/product/_reviews.html.twig
@@ -27,6 +27,6 @@
         <button class="btn btn-primary" formnovalidate>Add Review</button>
     {{ form_end(reviewForm) }}
 {% elseif not is_granted('ROLE_USER') %}
-    <p><a href="{{ path('app_login') }}">Log in</a> to post your review</p>
+    <p><a href="{{ path('app_login') }}" data-turbo-frame="_top">Log in</a> to post your review</p>
 {% endif %}
 </turbo-frame>
