diff --git a/assets/controllers/search-preview_controller.js b/assets/controllers/search-preview_controller.js
index db5f97c..5fa9f03 100644
--- a/assets/controllers/search-preview_controller.js
+++ b/assets/controllers/search-preview_controller.js
@@ -6,6 +6,10 @@ export default class extends Controller {
     }
 
     onSearchInput(event) {
-        console.log(this.urlValue);
+        const params = new URLSearchParams({
+            q: event.currentTarget.value,
+            preview: 1,
+        });
+        fetch(`${this.urlValue}?${params.toString()}`);
     }
 }
