diff --git a/assets/controllers/search-preview_controller.js b/assets/controllers/search-preview_controller.js
new file mode 100644
index 0000000..66e3936
--- /dev/null
+++ b/assets/controllers/search-preview_controller.js
@@ -0,0 +1,7 @@
+import { Controller } from 'stimulus';
+
+export default class extends Controller {
+    connect() {
+        console.log('connected!');
+    }
+}
diff --git a/templates/product/index.html.twig b/templates/product/index.html.twig
index d44eb6a..5fd3df7 100644
--- a/templates/product/index.html.twig
+++ b/templates/product/index.html.twig
@@ -36,7 +36,10 @@
                     </div>
                     <div class="col-9">
                         <form>
-                            <div class="input-group">
+                            <div
+                                class="input-group"
+                                data-controller="search-preview"
+                            >
                                 <input
                                     name="q"
                                     value="{{ searchTerm }}"
