diff --git a/assets/controllers/search-preview_controller.js b/assets/controllers/search-preview_controller.js
index 66e3936..b91d717 100644
--- a/assets/controllers/search-preview_controller.js
+++ b/assets/controllers/search-preview_controller.js
@@ -1,7 +1,7 @@
 import { Controller } from 'stimulus';
 
 export default class extends Controller {
-    connect() {
-        console.log('connected!');
+    onSearchInput(event) {
+        console.log(event);
     }
 }
diff --git a/templates/product/index.html.twig b/templates/product/index.html.twig
index 00514db..f52c337 100644
--- a/templates/product/index.html.twig
+++ b/templates/product/index.html.twig
@@ -46,6 +46,7 @@
                                     placeholder="Search products..."
                                     type="search"
                                     class="form-control"
+                                    data-action="search-preview#onSearchInput"
                                 >
                             </div>
                         </form>
