diff --git a/assets/controllers/search-preview_controller.js b/assets/controllers/search-preview_controller.js
index 2817b5b..0dac106 100644
--- a/assets/controllers/search-preview_controller.js
+++ b/assets/controllers/search-preview_controller.js
@@ -2,6 +2,8 @@ import { Controller } from 'stimulus';
 import { useClickOutside, useDebounce } from 'stimulus-use';
 
 export default class extends Controller {
+    query = '';
+
     static values = {
         url: String,
     }
@@ -19,8 +21,13 @@ export default class extends Controller {
     }
 
     async search(query) {
+        if (!query || query === this.query) {
+            return;
+        }
+        this.query = query;
+
         const params = new URLSearchParams({
-            q: query,
+            q: this.query,
             preview: 1,
         });
         const response = await fetch(`${this.urlValue}?${params.toString()}`);
