diff --git a/assets/controllers/search-preview_controller.js b/assets/controllers/search-preview_controller.js
index b91d7172..db5f97cf 100644
--- a/assets/controllers/search-preview_controller.js
+++ b/assets/controllers/search-preview_controller.js
@@ -1,7 +1,11 @@
 import { Controller } from 'stimulus';
 
 export default class extends Controller {
+    static values = {
+        url: String,
+    }
+
     onSearchInput(event) {
-        console.log(event);
+        console.log(this.urlValue);
     }
 }
diff --git a/templates/product/index.html.twig b/templates/product/index.html.twig
index f52c3378..3d5a68f5 100644
--- a/templates/product/index.html.twig
+++ b/templates/product/index.html.twig
@@ -38,7 +38,9 @@
                         <form>
                             <div
                                 class="input-group"
-                                {{ stimulus_controller('search-preview') }}
+                                {{ stimulus_controller('search-preview', {
+                                    url: path('app_homepage')
+                                }) }}
                             >
                                 <input
                                     name="q"
