diff --git a/assets/controllers/search-preview_controller.js b/assets/controllers/search-preview_controller.js
index 25eb911..8afbd9c 100644
--- a/assets/controllers/search-preview_controller.js
+++ b/assets/controllers/search-preview_controller.js
@@ -5,6 +5,8 @@ export default class extends Controller {
         url: String,
     }
 
+    static targets = ['results'];
+
     async onSearchKeyUp(event) {
         const params = new URLSearchParams({
             q: event.currentTarget.value,
@@ -12,6 +14,6 @@ export default class extends Controller {
         });
         const response = await fetch(`${this.urlValue}?${params.toString()}`);
 
-        console.log(await response.text());
+        this.resultsTarget.innerHTML = await response.text();
     }
 }
diff --git a/templates/product/index.html.twig b/templates/product/index.html.twig
index 1b5e911..e868d73 100644
--- a/templates/product/index.html.twig
+++ b/templates/product/index.html.twig
@@ -49,6 +49,11 @@
                                     class="form-control"
                                     data-action="keyup->search-preview#onSearchKeyUp"
                                 >
+
+                                <div
+                                    class="search-preview"
+                                    data-search-preview-target="results"
+                                ></div>
                             </div>
                         </form>
                     </div>
