diff --git a/src/Controller/ProductController.php b/src/Controller/ProductController.php
index 84fe4c3e..484d46ab 100644
--- a/src/Controller/ProductController.php
+++ b/src/Controller/ProductController.php
@@ -26,6 +26,12 @@ class ProductController extends AbstractController
             $searchTerm
         );
 
+        if ($request->query->get('preview')) {
+            return $this->render('product/_searchPreview.html.twig', [
+                'products' => $products,
+            ]);
+        }
+
         return $this->render('product/index.html.twig', [
             'currentCategory' => $category,
             'categories' => $categoryRepository->findAll(),
diff --git a/templates/product/_searchPreview.html.twig b/templates/product/_searchPreview.html.twig
new file mode 100644
index 00000000..131e1b6c
--- /dev/null
+++ b/templates/product/_searchPreview.html.twig
@@ -0,0 +1,12 @@
+<div class="list-group">
+    {% for product in products %}
+        <a
+            class="list-group-item list-group-item-action"
+            href="{{ path('app_product', { id: product.id }) }}"
+        >
+            {{ product.name }}
+        </a>
+    {% else %}
+        <div class="list-group-item">No results found!</div>
+    {% endfor %}
+</div>
