diff --git a/src/Controller/CartController.php b/src/Controller/CartController.php
index 465ff7b..778dd96 100644
--- a/src/Controller/CartController.php
+++ b/src/Controller/CartController.php
@@ -33,7 +33,7 @@ class CartController extends AbstractController
         ]);
     }
 
-    public function cartFeaturedProduct(ProductRepository $productRepository, Request $request): Response
+    public function cartFeaturedProduct(ProductRepository $productRepository, Request $request, CartStorage $cartStorage): Response
     {
         $featuredProduct = $productRepository->findFeatured();
         $addToCartForm = $this->createForm(AddItemToCartFormType::class, null, [
@@ -44,6 +44,7 @@ class CartController extends AbstractController
             'featuredProduct' => $featuredProduct,
             'addToCartForm' => $addToCartForm,
             'showDescription' => $request->query->get('description'),
+            'cart' => $cartStorage->getOrCreateCart(),
         ]);
     }
 
