diff --git a/src/Controller/CartController.php b/src/Controller/CartController.php
index 83079bae..1c1b443f 100644
--- a/src/Controller/CartController.php
+++ b/src/Controller/CartController.php
@@ -28,6 +28,9 @@ class CartController extends AbstractController
         ]);
     }
 
+    /**
+     * @Route("/cart/_featured", name="_app_cart_product_featured")
+     */
     public function _cartFeaturedProduct(ProductRepository $productRepository): Response
     {
         $featuredProduct = $productRepository->findFeatured();
diff --git a/templates/cart/cart.html.twig b/templates/cart/cart.html.twig
index bd3f89ac..a5196bf4 100644
--- a/templates/cart/cart.html.twig
+++ b/templates/cart/cart.html.twig
@@ -9,7 +9,7 @@
     <div class="container-fluid container-xl mt-4">
         <div class="row">
             <aside class="col-12 col-md-4 order-2 order-md-1">
-                <turbo-frame id="cart-sidebar" src="{{ fragment_uri(controller('App\\Controller\\CartController::_cartFeaturedProduct')) }}" target="_top">
+                <turbo-frame id="cart-sidebar" src="{{ path('_app_cart_product_featured') }}" target="_top">
                     Loading...
                 </turbo-frame>
             </aside>
