diff --git a/templates/cart/cart.html.twig b/templates/cart/cart.html.twig
index a5196bf4..de4b433f 100644
--- a/templates/cart/cart.html.twig
+++ b/templates/cart/cart.html.twig
@@ -9,9 +9,13 @@
     <div class="container-fluid container-xl mt-4">
         <div class="row">
             <aside class="col-12 col-md-4 order-2 order-md-1">
-                <turbo-frame id="cart-sidebar" src="{{ path('_app_cart_product_featured') }}" target="_top">
-                    Loading...
-                </turbo-frame>
+                {% if featuredProduct is defined %}
+                    {{ include('cart/_featuredSidebar.html.twig') }}
+                {% else %}
+                    <turbo-frame id="cart-sidebar" src="{{ path('_app_cart_product_featured') }}" target="_top">
+                        Loading...
+                    </turbo-frame>
+                {% endif %}
             </aside>
 
             <div class="col-12 col-md-8 order-1 order-md-2 cart">
