diff --git a/templates/cart/_featuredSidebar.html.twig b/templates/cart/_featuredSidebar.html.twig
index da22a06..99d777b 100644
--- a/templates/cart/_featuredSidebar.html.twig
+++ b/templates/cart/_featuredSidebar.html.twig
@@ -15,6 +15,8 @@
                 {{ featuredProduct.priceString|format_currency('USD') }}
             </h6>
 
+            {{ featuredProduct.description|u.truncate(25)|trim }}...
+
             {{ include('product/_cart_add_controls.html.twig') }}
         </div>
     </div>
