diff --git a/src/Controller/CartController.php b/src/Controller/CartController.php
index 5b1f6767..0b9dbe82 100644
--- a/src/Controller/CartController.php
+++ b/src/Controller/CartController.php
@@ -44,7 +44,7 @@ class CartController extends AbstractController
             'product' => $featuredProduct,
         ]);
 
-        return $this->renderForm('cart/_featuredSidebar.html.twig', [
+        return $this->renderForm('cart/cart.html.twig', [
             'featuredProduct' => $featuredProduct,
             'addToCartForm' => $addToCartForm,
             'showDescription' => $request->query->get('description'),
