diff --git a/src/Controller/CartController.php b/src/Controller/CartController.php
index 9cc3567b..465ff7be 100644
--- a/src/Controller/CartController.php
+++ b/src/Controller/CartController.php
@@ -40,7 +40,7 @@ class CartController extends AbstractController
             'product' => $featuredProduct,
         ]);
 
-        return $this->renderForm('cart/_featuredSidebar.html.twig', [
+        return $this->renderForm('cart/cart.html.twig', [
             'featuredProduct' => $featuredProduct,
             'addToCartForm' => $addToCartForm,
             'showDescription' => $request->query->get('description'),
