diff --git a/assets/app.js b/assets/app.js
index 2ce8f0e..6a303d4 100644
--- a/assets/app.js
+++ b/assets/app.js
@@ -26,7 +26,7 @@ document.addEventListener('turbo:before-cache', () => {
     // internal way to see if sweetalert2 has been imported yet
     if (__webpack_modules__[require.resolveWeak('sweetalert2')]) {
         // because we know it's been imported, this will run synchronously
-        import('sweetalert2').then((Swal) => {
+        import(/* webpackMode: 'weak' */'sweetalert2').then((Swal) => {
             if (Swal.default.isVisible()) {
                 Swal.default.getPopup().style.animationDuration = '0ms'
                 Swal.default.close();
