diff --git a/assets/app.js b/assets/app.js
index 2c9c02cb..1a824994 100644
--- a/assets/app.js
+++ b/assets/app.js
@@ -15,7 +15,11 @@ import { Modal } from 'bootstrap';
 
 document.addEventListener('turbo:before-cache', () => {
     if (document.body.classList.contains('modal-open')) {
-        const modal = Modal.getInstance(document.querySelector('.modal'));
+        const modalEl = document.querySelector('.modal');
+        const modal = Modal.getInstance(modalEl);
+        modalEl.classList.remove('fade');
+        modal._backdrop._config.isAnimated = false;
         modal.hide();
+        modal.dispose();
     }
 });
