diff --git a/assets/app.js b/assets/app.js
index 1a8249946..5cfe95a53 100644
--- a/assets/app.js
+++ b/assets/app.js
@@ -12,6 +12,7 @@ import './styles/app.css';
 import './bootstrap';
 
 import { Modal } from 'bootstrap';
+import Swal from 'sweetalert2';
 
 document.addEventListener('turbo:before-cache', () => {
     if (document.body.classList.contains('modal-open')) {
@@ -22,4 +23,8 @@ document.addEventListener('turbo:before-cache', () => {
         modal.hide();
         modal.dispose();
     }
+
+    if (Swal.isVisible()) {
+        Swal.close();
+    }
 });
