diff --git a/assets/controllers/search-preview_controller.js b/assets/controllers/search-preview_controller.js
index a298f097..d0af6df5 100644
--- a/assets/controllers/search-preview_controller.js
+++ b/assets/controllers/search-preview_controller.js
@@ -1,4 +1,5 @@
 import { Controller } from 'stimulus';
+import { useClickOutside } from 'stimulus-use';
 
 export default class extends Controller {
     static values = {
@@ -7,6 +8,10 @@ export default class extends Controller {
 
     static targets = ['result'];
 
+    connect() {
+        useClickOutside(this);
+    }
+
     async onSearchInput(event) {
         const params = new URLSearchParams({
             q: event.currentTarget.value,
@@ -16,4 +21,8 @@ export default class extends Controller {
 
         this.resultTarget.innerHTML = await response.text();
     }
+
+    clickOutside(event) {
+        this.resultTarget.innerHTML = '';
+    }
 }
