diff --git a/assets/controllers/search-preview_controller.js b/assets/controllers/search-preview_controller.js
index f00b147..2817b5b 100644
--- a/assets/controllers/search-preview_controller.js
+++ b/assets/controllers/search-preview_controller.js
@@ -7,16 +7,20 @@ export default class extends Controller {
     }
 
     static targets = ['result'];
-    static debounces = ['onSearchInput'];
+    static debounces = ['search'];
 
     connect() {
         useClickOutside(this);
         useDebounce(this);
     }
 
-    async onSearchInput(event) {
+    onSearchInput(event) {
+        this.search(event.currentTarget.value);
+    }
+
+    async search(query) {
         const params = new URLSearchParams({
-            q: event.currentTarget.value,
+            q: query,
             preview: 1,
         });
         const response = await fetch(`${this.urlValue}?${params.toString()}`);
