diff --git a/src/Controller/ProductController.php b/src/Controller/ProductController.php
index e32023c1..de429c3c 100644
--- a/src/Controller/ProductController.php
+++ b/src/Controller/ProductController.php
@@ -87,10 +87,13 @@ class ProductController extends AbstractController
                 $entityManager->persist($reviewForm->getData());
                 $entityManager->flush();
 
+                $message = 'Thanks for your review! I like you!';
+
                 return $this->render('product/reviews.stream.html.twig', [
+                    'message' => $message,
                 ], new TurboStreamResponse());
 
-                $this->addFlash('success', 'Thanks for your review! I like you!');
+                $this->addFlash('success', $message);
 
                 return $this->redirectToRoute('app_product_reviews', [
                     'id' => $product->getId(),
diff --git a/templates/product/reviews.stream.html.twig b/templates/product/reviews.stream.html.twig
index 513fb6b9..74e593ef 100644
--- a/templates/product/reviews.stream.html.twig
+++ b/templates/product/reviews.stream.html.twig
@@ -1,5 +1,8 @@
 <turbo-stream action="append" target="flash-container">
     <template>
-        Will this <strong>really</strong> work???
+        {{ include('_flash.html.twig', {
+            type: 'success',
+            message: message,
+        }) }}
     </template>
 </turbo-stream>
