diff --git a/assets/controllers/streamed-item_controller.js b/assets/controllers/streamed-item_controller.js
index d372200..d20b7be 100644
--- a/assets/controllers/streamed-item_controller.js
+++ b/assets/controllers/streamed-item_controller.js
@@ -7,5 +7,9 @@ export default class extends Controller {
 
     connect() {
         this.element.classList.add(this.classNameValue);
+        setTimeout(() => {
+            this.element.classList.add('fade-background');
+            this.element.classList.remove(this.classNameValue);
+        }, 5000);
     }
 }
