diff --git a/templates/product/_review.html.twig b/templates/product/_review.html.twig
new file mode 100644
index 000000000..fc843f1b9
--- /dev/null
+++ b/templates/product/_review.html.twig
@@ -0,0 +1,6 @@
+<div class="component-light my-3 p-3">
+    <p><i class="fas fa-user-circle me-2"></i>{{ review.owner.email }} <i class="fas fa-star ms-4"></i> {{ review.stars }}/5</p>
+    <div>
+        {{ review.content }}
+    </div>
+</div>
diff --git a/templates/product/_reviews_list.html.twig b/templates/product/_reviews_list.html.twig
index 05630227f..ba66fe9ce 100644
--- a/templates/product/_reviews_list.html.twig
+++ b/templates/product/_reviews_list.html.twig
@@ -1,11 +1,6 @@
 <div id="product-{{ product.id }}-review-list">
 {% for review in product.reviews %}
-    <div class="component-light my-3 p-3">
-        <p><i class="fas fa-user-circle me-2"></i>{{ review.owner.email }} <i class="fas fa-star ms-4"></i> {{ review.stars }}/5</p>
-        <div>
-            {{ review.content }}
-        </div>
-    </div>
+    {{ include('product/_review.html.twig') }}
 {% else %}
     <p>This product has not been reviewed yet!</p>
 {% endfor %}
