diff --git a/templates/product/reviews.stream.html.twig b/templates/product/reviews.stream.html.twig
index a30620e..a1fa9c1 100644
--- a/templates/product/reviews.stream.html.twig
+++ b/templates/product/reviews.stream.html.twig
@@ -1,5 +1,5 @@
 <turbo-stream action="update" target="product-quick-stats">
     <template>
-        Will this <strong>really</strong> work???
+        {{ include('product/_quickStats.html.twig') }}
     </template>
 </turbo-stream>
