diff --git a/templates/product/reviews.stream.html.twig b/templates/product/reviews.stream.html.twig
index 4f965da3a..e62244266 100644
--- a/templates/product/reviews.stream.html.twig
+++ b/templates/product/reviews.stream.html.twig
@@ -6,6 +6,6 @@
 
 <turbo-stream action="replace" target="product-{{ product.id }}-review">
     <template>
-        {{ include('product/_reviews.html.twig') }}
+        {{ include('product/_reviews_list.html.twig') }}
     </template>
 </turbo-stream>
