diff --git a/templates/product/reviews.stream.html.twig b/templates/product/reviews.stream.html.twig
index 058c7e56..f2d072df 100644
--- a/templates/product/reviews.stream.html.twig
+++ b/templates/product/reviews.stream.html.twig
@@ -12,3 +12,9 @@
         {{ include('product/_reviews.html.twig') }}
     </template>
 </turbo-stream>
+
+<turbo-stream action="update" target="product-quick-stats">
+    <template>
+        {{ include('product/_quickStats.html.twig') }}
+    </template>
+</turbo-stream>
