diff --git a/src/Controller/ProductController.php b/src/Controller/ProductController.php
index 479892e0e..ffab1e232 100644
--- a/src/Controller/ProductController.php
+++ b/src/Controller/ProductController.php
@@ -88,6 +88,7 @@ class ProductController extends AbstractController
                 $entityManager->flush();
 
                 return $this->render('product/reviews.stream.html.twig', [
+                    'product' => $product,
                 ], new TurboStreamResponse());
 
                 $this->addFlash('review_success', 'Thanks for your review! I like you!');
