diff --git a/templates/product/_reviews.html.twig b/templates/product/_reviews.html.twig
index 18411e3..bff616a 100644
--- a/templates/product/_reviews.html.twig
+++ b/templates/product/_reviews.html.twig
@@ -11,9 +11,12 @@
 {% else %}
     <p>This product has not been reviewed yet!</p>
 {% endfor %}
+</turbo-frame>
 
 <hr>
 
+<turbo-frame id="product-reviews-form">
+
 {% for flash in app.flashes('review_success') %}
     <div class="alert alert-success">{{ flash }}</div>
 {% endfor %}
