diff --git a/templates/product/reviews.stream.html.twig b/templates/product/reviews.stream.html.twig
index 2dbb3670..a1fa9c10 100644
--- a/templates/product/reviews.stream.html.twig
+++ b/templates/product/reviews.stream.html.twig
@@ -3,9 +3,3 @@
         {{ include('product/_quickStats.html.twig') }}
     </template>
 </turbo-stream>
-
-<turbo-stream action="replace" target="product-review">
-    <template>
-        {{ include('product/_reviews.html.twig') }}
-    </template>
-</turbo-stream>
