diff --git a/src/Controller/ProductController.php b/src/Controller/ProductController.php
index 73e655a..309e18f 100644
--- a/src/Controller/ProductController.php
+++ b/src/Controller/ProductController.php
@@ -91,7 +91,7 @@ class ProductController extends AbstractController
 
                 $update = new Update(
                     'product-reviews',
-                    $this->render('product/reviews.stream.html.twig', [
+                    $this->renderView('product/reviews.stream.html.twig', [
                         'product' => $product,
                     ]),
                 );
