diff --git a/assets/controllers/submit-confirm_controller.js b/assets/controllers/submit-confirm_controller.js
index ee62288..be60b0a 100644
--- a/assets/controllers/submit-confirm_controller.js
+++ b/assets/controllers/submit-confirm_controller.js
@@ -7,6 +7,7 @@ export default class extends Controller {
         text: String,
         icon: String,
         confirmButtonText: String,
+        submitAsync: Boolean,
     }
 
     onSubmit(event) {
@@ -28,6 +29,12 @@ export default class extends Controller {
     }
 
     submitForm() {
+        if (!this.submitAsyncValue) {
+            this.element.submit();
+
+            return;
+        }
+
         return fetch(this.element.action, {
             method: this.element.method,
             body: new URLSearchParams(new FormData(this.element)),
