diff --git a/assets/controllers/submit-confirm_controller.js b/assets/controllers/submit-confirm_controller.js
new file mode 100644
index 000000000..f7862aba1
--- /dev/null
+++ b/assets/controllers/submit-confirm_controller.js
@@ -0,0 +1,7 @@
+import { Controller } from 'stimulus';
+
+export default class extends Controller {
+    connect() {
+        console.log('🦖');
+    }
+}
diff --git a/templates/cart/cart.html.twig b/templates/cart/cart.html.twig
index 7f716513b..f1229d137 100644
--- a/templates/cart/cart.html.twig
+++ b/templates/cart/cart.html.twig
@@ -48,6 +48,7 @@
                                             colorId: item.color ? item.color.id: null,
                                         }) }}"
                                         method="POST"
+                                        {{ stimulus_controller('submit-confirm') }}
                                     >
                                         <input type="hidden" name="_token" value="{{ csrf_token('remove_item') }}" />
 
