diff --git a/templates/cart/cart.html.twig b/templates/cart/cart.html.twig
index 27599a5f..57bdc0e8 100644
--- a/templates/cart/cart.html.twig
+++ b/templates/cart/cart.html.twig
@@ -51,7 +51,8 @@
                                         {{ stimulus_controller('submit-confirm', {
                                             title: 'Remove this item?',
                                             icon: 'warning',
-                                            confirmButtonText: 'Yes, remove it'
+                                            confirmButtonText: 'Yes, remove it',
+                                            submitAsync: true,
                                         }) }}
                                         data-action="submit-confirm#onSubmit"
                                     >
