diff --git a/assets/controllers/submit-confirm_controller.js b/assets/controllers/submit-confirm_controller.js
index 7e021c0..76891a5 100644
--- a/assets/controllers/submit-confirm_controller.js
+++ b/assets/controllers/submit-confirm_controller.js
@@ -4,6 +4,7 @@ import Swal from 'sweetalert2';
 export default class extends Controller {
     onSubmit(event) {
         event.preventDefault();
+        const formElement = event.currentTarget;
 
         Swal.fire({
             title: 'Are you sure?',
@@ -15,15 +16,15 @@ export default class extends Controller {
             confirmButtonText: 'Yes, remove it!',
             showLoaderOnConfirm: true,
             preConfirm: () => {
-                return this.removeFromCart();
+                return this.removeFromCart(formElement);
             }
         });
     }
 
-    async removeFromCart() {
-        await fetch(this.element.action, {
-            method: this.element.method,
-            body: new URLSearchParams(new FormData(this.element)),
+    async removeFromCart(formElement) {
+        await fetch(formElement.action, {
+            method: formElement.method,
+            body: new URLSearchParams(new FormData(formElement)),
         });
 
         this.element.remove();
diff --git a/templates/cart/cart.html.twig b/templates/cart/cart.html.twig
index 915bc80..09f6751 100644
--- a/templates/cart/cart.html.twig
+++ b/templates/cart/cart.html.twig
@@ -26,7 +26,10 @@
                             <div class="col-3"></div>
                         </div>
                         {% for item in cart.items %}
-                            <div class="cart-item row p-3">
+                            <div
+                                class="cart-item row p-3"
+                                data-controller="submit-confirm"
+                            >
                                 <div class="col-2">
                                     {{ item.product.name }}
                                 </div>
@@ -48,7 +51,6 @@
                                             colorId: item.color ? item.color.id: null,
                                         }) }}"
                                         method="POST"
-                                        data-controller="submit-confirm"
                                         data-action="submit-confirm#onSubmit"
                                     >
                                         <input type="hidden" name="_token" value="{{ csrf_token('remove_item') }}" />
