diff --git a/assets/controllers/submit-confirm_controller.js b/assets/controllers/submit-confirm_controller.js
index fcc4eea..c3c53b4 100644
--- a/assets/controllers/submit-confirm_controller.js
+++ b/assets/controllers/submit-confirm_controller.js
@@ -24,10 +24,12 @@ export default class extends Controller {
         });
     }
 
-    removeFromCart() {
-        return fetch(this.element.action, {
+    async removeFromCart() {
+        await fetch(this.element.action, {
             method: this.element.method,
             body: new URLSearchParams(new FormData(this.element)),
         });
+
+        this.element.remove();
     }
 }
