diff --git a/assets/controllers/submit-confirm_controller.js b/assets/controllers/submit-confirm_controller.js
index f7862ab..5ea0ca2 100644
--- a/assets/controllers/submit-confirm_controller.js
+++ b/assets/controllers/submit-confirm_controller.js
@@ -1,7 +1,8 @@
 import { Controller } from 'stimulus';
 
 export default class extends Controller {
-    connect() {
-        console.log('🦖');
+    onSubmit(event) {
+        event.preventDefault();
+        console.log(event);
     }
 }
diff --git a/templates/cart/cart.html.twig b/templates/cart/cart.html.twig
index f1229d1..e2476da 100644
--- a/templates/cart/cart.html.twig
+++ b/templates/cart/cart.html.twig
@@ -49,6 +49,7 @@
                                         }) }}"
                                         method="POST"
                                         {{ stimulus_controller('submit-confirm') }}
+                                        data-action="submit-confirm#onSubmit"
                                     >
                                         <input type="hidden" name="_token" value="{{ csrf_token('remove_item') }}" />
 
