diff --git a/assets/controllers/toast_controller.js b/assets/controllers/toast_controller.js
index 87ff1ee79..a94edbe93 100644
--- a/assets/controllers/toast_controller.js
+++ b/assets/controllers/toast_controller.js
@@ -1,7 +1,9 @@
 import { Controller } from 'stimulus';
+import { Toast } from 'bootstrap';
 
 export default class extends Controller {
     connect() {
-        console.log('🍞');
+        const toast = new Toast(this.element);
+        toast.show();
     }
 }
