diff --git a/assets/turbo/turbo-helper.js b/assets/turbo/turbo-helper.js
index 08b6f22b..3b7b48ff 100644
--- a/assets/turbo/turbo-helper.js
+++ b/assets/turbo/turbo-helper.js
@@ -15,16 +15,32 @@ const TurboHelper = class {
             // fade out the old body
             document.body.classList.add('turbo-loading');
         });
+
         document.documentElement.addEventListener('turbo:before-render', (event) => {
-            // when we are *about* to render, start us faded out
-            event.detail.newBody.classList.add('turbo-loading');
+            if (this.isPreviewRendered()) {
+                // this is a preview that has been instantly swapped
+                // remove .turbo-loading so the preview starts fully opaque
+                event.detail.newBody.classList.remove('turbo-loading');
+                // start fading out 10ms later after opacity starts full
+                setTimeout(() => {
+                    document.body.classList.add('turbo-loading');
+                }, 10);
+            } else {
+                // when we are *about* to render a fresh page
+                // we should already be faded out, so start us faded out
+                event.detail.newBody.classList.add('turbo-loading');
+            }
         });
         document.documentElement.addEventListener('turbo:render', () => {
-            // after rendering, we first allow the .turbo-loaded to set the low opacity
-            // THEN, 10ms later, we remove the turbo-loaded class, which allows the fade in
-            setTimeout(() => {
-                document.body.classList.remove('turbo-loading');
-            }, 10);
+            if (!this.isPreviewRendered()) {
+                // if this is a preview, then we do nothing: stay faded out
+                // after rendering the REAL page, we first allow the .turbo-loading to
+                // instantly start the page at lower opacity. THEN remove the class,
+                // which allows the fade in
+                setTimeout(() => {
+                    document.body.classList.remove('turbo-loading');
+                }, 10);
+            }
         });
     }
 
@@ -55,6 +71,10 @@ const TurboHelper = class {
     initializeWeatherWidget() {
         __weatherwidget_init();
     }
+
+    isPreviewRendered() {
+        return document.documentElement.hasAttribute('data-turbo-preview');
+    }
 }
 
 export default new TurboHelper();
