diff --git a/templates/product/_cart_add_controls.html.twig b/templates/product/_cart_add_controls.html.twig
index 781e52c..f0b0e90 100644
--- a/templates/product/_cart_add_controls.html.twig
+++ b/templates/product/_cart_add_controls.html.twig
@@ -3,8 +3,9 @@
 }) }}
     {% if addToCartForm.color is defined %}
         <div
-            data-controller="color-square"
-            data-color-square-color-id-value="{{ addToCartForm.vars.data.product.colors[1].id }}"
+            {{ stimulus_controller('color-square', {
+                'color-id': addToCartForm.vars.data.product.colors[1].id
+            }) }}
         >
             {{ form_widget(addToCartForm.color, {
                 attr: { 'data-color-square-target': 'select' }
