diff --git a/assets/controllers/color-square_controller.js b/assets/controllers/color-square_controller.js
index 67abb74..863c065 100644
--- a/assets/controllers/color-square_controller.js
+++ b/assets/controllers/color-square_controller.js
@@ -44,6 +44,6 @@ export default class extends Controller {
      * @return {Element|null}
      */
     findSelectedColorSquare() {
-        return this.colorSquareTargets.find((element) => element.dataset.colorId === this.selectedColorId);
+        return this.colorSquareTargets.find((element) => element.dataset.colorId == this.selectedColorId);
     }
 }
