diff --git a/assets/controllers/color-square_controller.js b/assets/controllers/color-square_controller.js
index 8785abf..67abb74 100644
--- a/assets/controllers/color-square_controller.js
+++ b/assets/controllers/color-square_controller.js
@@ -11,7 +11,9 @@ export default class extends Controller {
     connect() {
         this.selectTarget.classList.add('d-none');
 
-        console.log(this.colorIdValue);
+        if (this.colorIdValue) {
+            this.setSelectedColor(this.colorIdValue);
+        }
     }
 
     selectColor(event) {
