diff --git a/assets/app.js b/assets/app.js
index 082087e6..031f9967 100644
--- a/assets/app.js
+++ b/assets/app.js
@@ -35,6 +35,6 @@ document.addEventListener('turbo:before-cache', () => {
     }
 });
 
-document.addEventListener('turbo:before-render', () => {
-    document.querySelector('#weatherwidget-io-js').remove();
+document.addEventListener('turbo:render', () => {
+    __weatherwidget_init();
 });
