diff --git a/assets/turbo/turbo-helper.js b/assets/turbo/turbo-helper.js
index e78b2663f..dcefc45fc 100644
--- a/assets/turbo/turbo-helper.js
+++ b/assets/turbo/turbo-helper.js
@@ -8,7 +8,7 @@ const TurboHelper = class {
         });
 
         document.addEventListener('turbo:render', () => {
-            __weatherwidget_init();
+            this.initializeWeatherWidget();
         });
     }
 
@@ -35,6 +35,10 @@ const TurboHelper = class {
             })
         }
     }
+
+    initializeWeatherWidget() {
+        __weatherwidget_init();
+    }
 }
 
 export default new TurboHelper();
