diff --git a/assets/util/turbo-helper.js b/assets/util/turbo-helper.js
index ab98bd99..06e64bb1 100644
--- a/assets/util/turbo-helper.js
+++ b/assets/util/turbo-helper.js
@@ -8,7 +8,7 @@ const TurboHelper = class {
         });
 
         document.addEventListener('turbo:before-render', () => {
-            console.log('before render!');
+            document.querySelector('#weatherwidget-io-js').remove();
         });
     }
 
