diff --git a/assets/turbo/turbo-helper.js b/assets/turbo/turbo-helper.js
index 7406f98d2..ab98bd991 100644
--- a/assets/turbo/turbo-helper.js
+++ b/assets/turbo/turbo-helper.js
@@ -6,6 +6,10 @@ const TurboHelper = class {
             this.closeModal();
             this.closeSweetalert();
         });
+
+        document.addEventListener('turbo:before-render', () => {
+            console.log('before render!');
+        });
     }
 
     closeModal() {
